<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>machine_learning_analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="machine_learning_analysis_files/libs/clipboard/clipboard.min.js"></script>
<script src="machine_learning_analysis_files/libs/quarto-html/quarto.js"></script>
<script src="machine_learning_analysis_files/libs/quarto-html/popper.min.js"></script>
<script src="machine_learning_analysis_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="machine_learning_analysis_files/libs/quarto-html/anchor.min.js"></script>
<link href="machine_learning_analysis_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="machine_learning_analysis_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="machine_learning_analysis_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="machine_learning_analysis_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="machine_learning_analysis_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">machine_learning_analysis</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="machine-learning-analysis" class="level1">
<h1>Machine Learning Analysis</h1>
<p>Here we will be performing a machine learning analysis of our proteomics and metabolomics data.</p>
<p>We intend to evaluate several models focusing on 3 comparisons:</p>
<ol type="1">
<li>Healthy vs Infected patients</li>
<li>E.faecalis vs E.faecium infected patients</li>
<li>Mortality vs Surival.</li>
</ol>
<p>Our dataset has low n but high p, so we will apply a lasso regression in hopes that this will prevent overfitting of the data.</p>
<p>We will split our</p>
<section id="loadingformatting-data" class="level2">
<h2 class="anchored" data-anchor-id="loadingformatting-data">Loading/Formatting Data</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    filter, lag</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidymodels)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching packages ────────────────────────────────────── tidymodels 1.2.0 ──</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>✔ broom        1.0.7     ✔ rsample      1.2.1
✔ dials        1.3.0     ✔ tibble       3.2.1
✔ ggplot2      3.5.1     ✔ tidyr        1.3.1
✔ infer        1.0.7     ✔ tune         1.2.1
✔ modeldata    1.4.0     ✔ workflows    1.1.4
✔ parsnip      1.2.1     ✔ workflowsets 1.1.0
✔ purrr        1.0.2     ✔ yardstick    1.3.1
✔ recipes      1.1.0     </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Conflicts ───────────────────────────────────────── tidymodels_conflicts() ──
✖ purrr::discard() masks scales::discard()
✖ dplyr::filter()  masks stats::filter()
✖ dplyr::lag()     masks stats::lag()
✖ recipes::step()  masks stats::step()
• Search for functions across packages at https://www.tidymodels.org/find/</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(caret)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: lattice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'caret'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:yardstick':

    precision, recall, sensitivity, specificity</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    lift</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(broom)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'readr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:yardstick':

    spec</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scales':

    col_factor</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(thematic)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>ggplot2<span class="sc">::</span><span class="fu">theme_set</span>(<span class="fu">theme_bw</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-fig.showtext="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>clinical_metadata <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"clinical_metadata.csv"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 105 Columns: 75
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (29): sample_id, patient_id, gender, race, liver_disease, diabetes, soli...
dbl (22): day_of_blood_draw, age, days_in_hosptial, charleson_index, bmi, ba...
lgl (24): death_during_admission, death_at_one_year, history_of_myocardial_i...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Clinical metadata with only the relevant outcomes</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>clinical_metadata_o <span class="ot">=</span> clinical_metadata <span class="sc">%&gt;%</span> </span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(sample_id, condition, infection_status,</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>           death_during_admission)</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>proteomics_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">'proteomics_data_cleaned.csv'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 273 Columns: 109
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr   (4): orientation, uniprotid, locus, Gene
dbl (105): S1, S3, S2, S5, S4, S7, S6, S9, S8, S11, S10, S13, S12, S15, S14,...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>metabolomics_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">'metabolomics_data_cleaned.csv'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 2404 Columns: 107
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr   (2): row_id, compound_name
dbl (105): S7, S2, S12, S16, S4, S15, S24, S17, S6, S19, S8, S9, S13, S1, S2...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
<div class="cell" data-fig.showtext="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Transposing proteomics</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>proteomics_t <span class="ot">&lt;-</span> proteomics_data <span class="sc">%&gt;%</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>orientation, <span class="sc">-</span>uniprotid, <span class="sc">-</span>locus) <span class="sc">%&gt;%</span> </span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(.),<span class="at">names_to =</span> <span class="st">"sample_id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> Gene, <span class="at">values_from =</span> value)</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Transposing metabolomics  </span></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>metabolomics_t <span class="ot">&lt;-</span> metabolomics_data <span class="sc">%&gt;%</span> </span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(compound_name)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">row_id =</span> <span class="fu">paste0</span>(row_id,<span class="st">":"</span>,compound_name)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>compound_name) <span class="sc">%&gt;%</span> </span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(.),<span class="at">names_to =</span> <span class="st">"sample_id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> row_id, <span class="at">values_from =</span> value)</span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Combining proteomics and metabolomics</span></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>all_t <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">inner_join</span>(proteomics_t,metabolomics_t,<span class="at">by =</span> <span class="st">"sample_id"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="model-function" class="level2">
<h2 class="anchored" data-anchor-id="model-function">Model Function</h2>
<p>Here we will define a function to do all the modeling steps. This way for each comparison, we only need to provide the model data and model recipie without repeating everything over and over.</p>
<div class="cell" data-fig.showtext="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Defining the model Max built as a function, so we can easily apply it to all </span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="co"># of our comparisons.</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>our_model <span class="ot">=</span> <span class="cf">function</span>(model_data,</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>                     outcome,</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>                     model_recipe){</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Setting seed at 42 to make reproducible.  </span></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Create train/test split</span></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>train_test_split <span class="ot">&lt;-</span> <span class="fu">initial_split</span>(model_data,<span class="at">prop =</span> <span class="fl">0.8</span>,</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">strata =</span> outcome)</span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>train_data <span class="ot">&lt;-</span> <span class="fu">training</span>(train_test_split)</span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>test_data <span class="ot">&lt;-</span> <span class="fu">testing</span>(train_test_split)</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Setting a logistic regression. </span></span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>log_reg <span class="ot">&lt;-</span> <span class="fu">logistic_reg</span>(</span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">mixture =</span> <span class="dv">1</span>, <span class="co"># Lasso regression</span></span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">penalty =</span> <span class="fu">tune</span>() <span class="co"># testing a few different penalty values</span></span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_engine</span>(<span class="st">"glmnet"</span>) </span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating a model workflow</span></span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a>model_workflow <span class="ot">&lt;-</span> <span class="fu">workflow</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_recipe</span>(model_recipe) <span class="sc">%&gt;%</span> </span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_model</span>(log_reg)</span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-26"><a href="#cb26-26" aria-hidden="true" tabindex="-1"></a><span class="co"># 5 fold cross validation</span></span>
<span id="cb26-27"><a href="#cb26-27" aria-hidden="true" tabindex="-1"></a>cv_folds <span class="ot">&lt;-</span> <span class="fu">vfold_cv</span>(train_data, <span class="at">v =</span> <span class="dv">5</span>)</span>
<span id="cb26-28"><a href="#cb26-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-29"><a href="#cb26-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Tune the penalty parameter</span></span>
<span id="cb26-30"><a href="#cb26-30" aria-hidden="true" tabindex="-1"></a>tune_res <span class="ot">&lt;-</span> <span class="fu">tune_grid</span>(</span>
<span id="cb26-31"><a href="#cb26-31" aria-hidden="true" tabindex="-1"></a>  model_workflow,</span>
<span id="cb26-32"><a href="#cb26-32" aria-hidden="true" tabindex="-1"></a>  <span class="at">resamples =</span> cv_folds,</span>
<span id="cb26-33"><a href="#cb26-33" aria-hidden="true" tabindex="-1"></a>  <span class="at">grid =</span> <span class="dv">5</span>,     <span class="co"># Try 5 values of the penalty</span></span>
<span id="cb26-34"><a href="#cb26-34" aria-hidden="true" tabindex="-1"></a>  <span class="at">metrics =</span> <span class="fu">metric_set</span>(roc_auc, accuracy), <span class="co"># generate performance metrics for roc_auc &amp; accuracy</span></span>
<span id="cb26-35"><a href="#cb26-35" aria-hidden="true" tabindex="-1"></a>  <span class="at">control =</span> <span class="fu">control_grid</span>(<span class="at">save_pred =</span> <span class="cn">TRUE</span>)</span>
<span id="cb26-36"><a href="#cb26-36" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb26-37"><a href="#cb26-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-38"><a href="#cb26-38" aria-hidden="true" tabindex="-1"></a><span class="co"># Show performance metrics and get the best model</span></span>
<span id="cb26-39"><a href="#cb26-39" aria-hidden="true" tabindex="-1"></a>best_result <span class="ot">&lt;-</span> tune_res <span class="sc">%&gt;%</span></span>
<span id="cb26-40"><a href="#cb26-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">collect_metrics</span>() <span class="sc">%&gt;%</span></span>
<span id="cb26-41"><a href="#cb26-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(.metric <span class="sc">==</span> <span class="st">"roc_auc"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb26-42"><a href="#cb26-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(mean, <span class="at">n =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb26-43"><a href="#cb26-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(<span class="dv">1</span>)</span>
<span id="cb26-44"><a href="#cb26-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-45"><a href="#cb26-45" aria-hidden="true" tabindex="-1"></a><span class="co"># Finalize the workflow using the model workflow and our best performing model</span></span>
<span id="cb26-46"><a href="#cb26-46" aria-hidden="true" tabindex="-1"></a>final_workflow <span class="ot">&lt;-</span> <span class="fu">finalize_workflow</span>(model_workflow, best_result)</span>
<span id="cb26-47"><a href="#cb26-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-48"><a href="#cb26-48" aria-hidden="true" tabindex="-1"></a><span class="co"># Fit the Final Model on the Training Data</span></span>
<span id="cb26-49"><a href="#cb26-49" aria-hidden="true" tabindex="-1"></a>fit_workflow <span class="ot">&lt;-</span> <span class="fu">fit</span>(final_workflow, train_data)</span>
<span id="cb26-50"><a href="#cb26-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-51"><a href="#cb26-51" aria-hidden="true" tabindex="-1"></a><span class="co"># Coefficients for features that seem to matter to the model, the rest have been set to 0 via lasso regression</span></span>
<span id="cb26-52"><a href="#cb26-52" aria-hidden="true" tabindex="-1"></a>model_summary <span class="ot">&lt;-</span> <span class="fu">tidy</span>(fit_workflow)</span>
<span id="cb26-53"><a href="#cb26-53" aria-hidden="true" tabindex="-1"></a>model_summary_out <span class="ot">=</span> model_summary <span class="sc">%&gt;%</span></span>
<span id="cb26-54"><a href="#cb26-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(estimate <span class="sc">!=</span> <span class="dv">0</span>)</span>
<span id="cb26-55"><a href="#cb26-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-56"><a href="#cb26-56" aria-hidden="true" tabindex="-1"></a> <span class="co"># Make predictions on the test data</span></span>
<span id="cb26-57"><a href="#cb26-57" aria-hidden="true" tabindex="-1"></a>  predictions <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit_workflow, test_data, <span class="at">type =</span> <span class="st">"prob"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb26-58"><a href="#cb26-58" aria-hidden="true" tabindex="-1"></a>    <span class="fu">bind_cols</span>(<span class="fu">predict</span>(fit_workflow, test_data)) <span class="sc">%&gt;%</span></span>
<span id="cb26-59"><a href="#cb26-59" aria-hidden="true" tabindex="-1"></a>    <span class="fu">bind_cols</span>(test_data)</span>
<span id="cb26-60"><a href="#cb26-60" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb26-61"><a href="#cb26-61" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Generate confusion matrix</span></span>
<span id="cb26-62"><a href="#cb26-62" aria-hidden="true" tabindex="-1"></a>  pred <span class="ot">&lt;-</span> predictions <span class="sc">%&gt;%</span> <span class="fu">select</span>(outcome) <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">as.factor</span>()</span>
<span id="cb26-63"><a href="#cb26-63" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> predictions <span class="sc">%&gt;%</span> <span class="fu">select</span>(.pred_class) <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">as.factor</span>()</span>
<span id="cb26-64"><a href="#cb26-64" aria-hidden="true" tabindex="-1"></a>  confusion_matrix <span class="ot">&lt;-</span> <span class="fu">confusionMatrix</span>(pred, res)</span>
<span id="cb26-65"><a href="#cb26-65" aria-hidden="true" tabindex="-1"></a>  confusion_matrix_out <span class="ot">&lt;-</span> confusion_matrix<span class="sc">$</span>table</span>
<span id="cb26-66"><a href="#cb26-66" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb26-67"><a href="#cb26-67" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plotting coefficients (Lasso-selected features)</span></span>
<span id="cb26-68"><a href="#cb26-68" aria-hidden="true" tabindex="-1"></a>  p1 <span class="ot">&lt;-</span> model_summary_out <span class="sc">%&gt;%</span> </span>
<span id="cb26-69"><a href="#cb26-69" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="fu">reorder</span>(term, estimate), estimate)) <span class="sc">+</span></span>
<span id="cb26-70"><a href="#cb26-70" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb26-71"><a href="#cb26-71" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>()</span>
<span id="cb26-72"><a href="#cb26-72" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb26-73"><a href="#cb26-73" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Return relevant outputs</span></span>
<span id="cb26-74"><a href="#cb26-74" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb26-75"><a href="#cb26-75" aria-hidden="true" tabindex="-1"></a>    <span class="at">model_summary =</span> model_summary_out,</span>
<span id="cb26-76"><a href="#cb26-76" aria-hidden="true" tabindex="-1"></a>    <span class="at">test_results =</span> predictions,</span>
<span id="cb26-77"><a href="#cb26-77" aria-hidden="true" tabindex="-1"></a>    <span class="at">model_summary_plot =</span> p1,</span>
<span id="cb26-78"><a href="#cb26-78" aria-hidden="true" tabindex="-1"></a>    <span class="at">confusion_matrix =</span> confusion_matrix_out</span>
<span id="cb26-79"><a href="#cb26-79" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb26-80"><a href="#cb26-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(out)</span>
<span id="cb26-81"><a href="#cb26-81" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb26-82"><a href="#cb26-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-83"><a href="#cb26-83" aria-hidden="true" tabindex="-1"></a><span class="co"># Extracting the ROC curves from the output of our_model()</span></span>
<span id="cb26-84"><a href="#cb26-84" aria-hidden="true" tabindex="-1"></a>extract_roc <span class="ot">=</span> <span class="cf">function</span>(list,outcome,prediction_col){</span>
<span id="cb26-85"><a href="#cb26-85" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb26-86"><a href="#cb26-86" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Convert the outcome variable to a factor</span></span>
<span id="cb26-87"><a href="#cb26-87" aria-hidden="true" tabindex="-1"></a>  predictions <span class="ot">&lt;-</span> list<span class="sc">$</span>test_results <span class="sc">%&gt;%</span> </span>
<span id="cb26-88"><a href="#cb26-88" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="sc">!!</span><span class="at">outcome :=</span> <span class="fu">as.factor</span>(<span class="sc">!!</span><span class="fu">sym</span>(outcome)))</span>
<span id="cb26-89"><a href="#cb26-89" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb26-90"><a href="#cb26-90" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Generate ROC curve data</span></span>
<span id="cb26-91"><a href="#cb26-91" aria-hidden="true" tabindex="-1"></a>  roc_data <span class="ot">&lt;-</span> <span class="fu">roc_curve</span>(predictions, <span class="sc">!!</span><span class="fu">sym</span>(outcome), <span class="sc">!!</span><span class="fu">sym</span>(prediction_col))</span>
<span id="cb26-92"><a href="#cb26-92" aria-hidden="true" tabindex="-1"></a>  roc_auc <span class="ot">&lt;-</span> <span class="fu">roc_auc</span>(predictions, <span class="sc">!!</span><span class="fu">sym</span>(outcome), <span class="sc">!!</span><span class="fu">sym</span>(prediction_col))</span>
<span id="cb26-93"><a href="#cb26-93" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb26-94"><a href="#cb26-94" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot ROC curve</span></span>
<span id="cb26-95"><a href="#cb26-95" aria-hidden="true" tabindex="-1"></a>  p1 <span class="ot">&lt;-</span> <span class="fu">autoplot</span>(roc_data) <span class="sc">+</span></span>
<span id="cb26-96"><a href="#cb26-96" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(<span class="st">"AUC = "</span>, <span class="fu">round</span>(roc_auc<span class="sc">$</span>.estimate, <span class="dv">2</span>)))</span>
<span id="cb26-97"><a href="#cb26-97" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb26-98"><a href="#cb26-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(p1)</span>
<span id="cb26-99"><a href="#cb26-99" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="healthy-vs-infected" class="level2">
<h2 class="anchored" data-anchor-id="healthy-vs-infected">Healthy vs Infected</h2>
<p>First, let’s see how well our model is able to predict healthy vs infected.</p>
<div class="cell" data-fig.showtext="false">

</div>
<div class="cell" data-fig.showtext="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>h_v_i_res <span class="ot">&lt;-</span> <span class="fu">our_model</span>(model_data,<span class="st">"infection_status"</span>,h_v_i_model_recipe)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Using an external vector in selections was deprecated in tidyselect 1.1.0.
ℹ Please use `all_of()` or `any_of()` instead.
  # Was:
  data %&gt;% select(outcome)

  # Now:
  data %&gt;% select(all_of(outcome))

See &lt;https://tidyselect.r-lib.org/reference/faq-external-vector.html&gt;.</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">extract_roc</span>(h_v_i_res,<span class="st">"infection_status"</span>,<span class="st">".pred_infected"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="machine_learning_analysis_files/figure-html/view%20ROC-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>Figure 1.</strong> Here we see that the model performs perfectly.</p>
<p>What features are important for this model?</p>
<div class="cell" data-fig.showtext="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>h_v_i_res<span class="sc">$</span>model_summary_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="machine_learning_analysis_files/figure-html/view%20estimates-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>Figure 2.</strong> Here we see the features that are important for the model. Several of these features are likely artifacts of the way the samples were collected. Phtalic anhydride and phthalate are plasticizers. These are almost certainly due to the fact that these samples were collected in different types fo plastic tubes. This was a point the reviewers brought up.</p>
<p>We should either run the model with these features removed, or entirely remove the metabolomics data from this comparison.</p>
<p>Also, note the value of the intercept. This is a reasonable value as there is some class imbalance here since we have 76 infected patients and only 29 uninfected.</p>
<div class="cell" data-fig.showtext="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Recipe does all of the feature engineering so that we don't have to do anything else to the input data</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>h_v_i_model_recipe2 <span class="ot">&lt;-</span> <span class="fu">recipe</span>(infection_status <span class="sc">~</span> ., <span class="at">data =</span> model_data) <span class="sc">%&gt;%</span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(sample_id, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(condition, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(<span class="st">`</span><span class="at">RID_29112:bis(2-ethylhexyl) phthalate</span><span class="st">`</span> , <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(<span class="st">`</span><span class="at">RID_22644:phthalic anhydride</span><span class="st">`</span> , <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(death_during_admission, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span>  <span class="co"># id role tells the model not to actually incorporate it as a feature</span></span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(infection_status, <span class="at">new_role =</span> <span class="st">"outcome"</span>) <span class="sc">%&gt;%</span> <span class="co"># specify target </span></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_dummy</span>(<span class="fu">all_nominal_predictors</span>()) <span class="sc">%&gt;%</span> <span class="co"># convert string to dummy variables</span></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_normalize</span>(<span class="fu">all_numeric_predictors</span>()) <span class="co"># normalize numeric features</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-fig.showtext="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>h_v_i_res2 <span class="ot">&lt;-</span> <span class="fu">our_model</span>(model_data,<span class="st">"infection_status"</span>,h_v_i_model_recipe2)</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="fu">extract_roc</span>(h_v_i_res2,<span class="st">"infection_status"</span>,<span class="st">".pred_infected"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="machine_learning_analysis_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>Figure 3.</strong> Model still performs perfectly with problematic metabolites removed.</p>
<div class="cell" data-fig.showtext="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>h_v_i_res2<span class="sc">$</span>model_summary_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="machine_learning_analysis_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>Figure 4.</strong> Here we can see the features important to this new model.</p>
</section>
<section id="e.-faecalis-vs-e.-faecium" class="level2">
<h2 class="anchored" data-anchor-id="e.-faecalis-vs-e.-faecium"><em>E. faecalis</em> vs <em>E. faecium</em></h2>
<p>Now let’s see how well the modeling performs for distinguishing <em>E.faecalis</em> from <em>E.faecium</em></p>
<div class="cell" data-fig.showtext="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Have to discard the healthy data here </span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>model_data <span class="ot">&lt;-</span> clinical_metadata_o <span class="sc">%&gt;%</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(all_t, <span class="at">by =</span> <span class="st">"sample_id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(condition <span class="sc">!=</span> <span class="st">"healthy"</span>)</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>outcome <span class="ot">=</span> <span class="st">"infection_status"</span></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Recipe does all of the feature engineering so that we don't have to do anything else to the input data</span></span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a>faecalis_v_faecium_recipe <span class="ot">&lt;-</span> <span class="fu">recipe</span>(condition <span class="sc">~</span> ., <span class="at">data =</span> model_data) <span class="sc">%&gt;%</span></span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(sample_id, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(infection_status, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(death_during_admission, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span>  <span class="co"># id role tells the model not to actually incorporate it as a feature</span></span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(condition, <span class="at">new_role =</span> <span class="st">"outcome"</span>) <span class="sc">%&gt;%</span> <span class="co"># specify target </span></span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_dummy</span>(<span class="fu">all_nominal_predictors</span>()) <span class="sc">%&gt;%</span> <span class="co"># convert string to dummy variables</span></span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_normalize</span>(<span class="fu">all_numeric_predictors</span>()) <span class="co"># normalize numeric features</span></span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a>faecalis_v_faecium_res <span class="ot">&lt;-</span> <span class="fu">our_model</span>(model_data,<span class="st">"condition"</span>,faecalis_v_faecium_recipe)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-fig.showtext="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">extract_roc</span>(faecalis_v_faecium_res,<span class="st">"condition"</span>,<span class="st">".pred_faecalis"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="machine_learning_analysis_files/figure-html/model%20ROCs-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>Figure 5.</strong> Here we can see the ROC curve for the E.facalis vs E.faecium prediction. The model looks like it is performing okay here. This isn’t good enough to be clinically useful really.</p>
<p>Note that a simple model where you assume everyone has E. faecalis gets it right 57% of the time since there are 43 patients with faecalis and only 32 with faecium.</p>
<div class="cell" data-fig.showtext="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>faecalis_v_faecium_res<span class="sc">$</span>model_summary_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="machine_learning_analysis_files/figure-html/model%20estimates-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>Figure 6.</strong> Here we can see the features that are most important to the model. Looks very similar to the previous analysis. Several of these proteins and metabolites are immune related.</p>
<p>Signal we are picking up here is probably that people with faecium infections are more likely to be immmunocompromised (this is already established in the literature.)</p>
<section id="metadata-alone-model" class="level3">
<h3 class="anchored" data-anchor-id="metadata-alone-model">Metadata alone model</h3>
<p>I am curious how well a model made with just the metadata would perform.</p>
<p>I would expect the MIC testing to do a good job of predicting faecium, as facieum isolates are more likely to be ampicillin and vancomycin resistant.</p>
<div class="cell" data-fig.showtext="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>model_data <span class="ot">&lt;-</span> clinical_metadata <span class="sc">%&gt;%</span> </span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(condition <span class="sc">!=</span> <span class="st">"healthy"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># antimicrobial testing was not perfomred for this patient</span></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sample_id <span class="sc">!=</span> <span class="st">"S18"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>  <span class="co">#getting rid of uninformative columns. Either very little variance, or too many levels to be useful </span></span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>pathogen,<span class="sc">-</span>patient_id,<span class="sc">-</span>pathogen_source,<span class="sc">-</span>infection_status,</span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>         <span class="sc">-</span>bacteremia_recurrence,<span class="sc">-</span>race,<span class="sc">-</span>death_at_one_year,</span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a>         <span class="sc">-</span>peripheral_vascular_disease,<span class="sc">-</span>dementia,<span class="sc">-</span>hemiplegia,</span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a>         <span class="sc">-</span>highly_active_antiretroviral_therapy,</span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a>         <span class="sc">-</span>source) <span class="sc">%&gt;%</span> </span>
<span id="cb37-11"><a href="#cb37-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">transplant_type =</span> <span class="fu">case_when</span>(transplant_type <span class="sc">==</span> <span class="st">"none"</span> <span class="sc">~</span> <span class="st">"none"</span>,</span>
<span id="cb37-12"><a href="#cb37-12" aria-hidden="true" tabindex="-1"></a>                                     <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"transplant"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb37-13"><a href="#cb37-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select_if</span>(<span class="sc">~</span> <span class="sc">!</span><span class="fu">any</span>(<span class="fu">is.na</span>(.))) <span class="sc">%&gt;%</span> </span>
<span id="cb37-14"><a href="#cb37-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>antibiotic_therapy_before_48_hrs,<span class="sc">-</span>antibiotic_therapy_after_48_hrs) <span class="sc">%&gt;%</span> </span>
<span id="cb37-15"><a href="#cb37-15" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(history_of_myocardial_infarction, </span>
<span id="cb37-16"><a href="#cb37-16" aria-hidden="true" tabindex="-1"></a>                   congestive_heart_failure, </span>
<span id="cb37-17"><a href="#cb37-17" aria-hidden="true" tabindex="-1"></a>                   history_of_stroke_or_transient_ischemic_attack, </span>
<span id="cb37-18"><a href="#cb37-18" aria-hidden="true" tabindex="-1"></a>                   smoking, copd, peptic_ulcer_disaese, </span>
<span id="cb37-19"><a href="#cb37-19" aria-hidden="true" tabindex="-1"></a>                   mod_to_severe_chromic_kidney_disease, </span>
<span id="cb37-20"><a href="#cb37-20" aria-hidden="true" tabindex="-1"></a>                   hodgkin_disease, leukemia_or_lymphoma, </span>
<span id="cb37-21"><a href="#cb37-21" aria-hidden="true" tabindex="-1"></a>                   icu_admission, mechanical_ventilation, </span>
<span id="cb37-22"><a href="#cb37-22" aria-hidden="true" tabindex="-1"></a>                    polymicrobial_bacteremia, </span>
<span id="cb37-23"><a href="#cb37-23" aria-hidden="true" tabindex="-1"></a>                   thrombocytopenia_plt_50, hypotension,sensitive_to_vancomycin), </span>
<span id="cb37-24"><a href="#cb37-24" aria-hidden="true" tabindex="-1"></a>                as.factor)) <span class="sc">%&gt;%</span> </span>
<span id="cb37-25"><a href="#cb37-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">wbc =</span> <span class="fu">as.numeric</span>(wbc))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `wbc = as.numeric(wbc)`.
Caused by warning:
! NAs introduced by coercion</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>faecalis_v_faecium_m_model_recipe <span class="ot">&lt;-</span> <span class="fu">recipe</span>(condition <span class="sc">~</span> ., <span class="at">data =</span> model_data) <span class="sc">%&gt;%</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(sample_id, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(death_during_admission, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span>  <span class="co"># id role tells the model not to actually incorporate it as a feature</span></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(condition, <span class="at">new_role =</span> <span class="st">"outcome"</span>) <span class="sc">%&gt;%</span> <span class="co"># specify target </span></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_novel</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_dummy</span>(<span class="fu">all_nominal_predictors</span>()) <span class="sc">%&gt;%</span> <span class="co"># convert string to dummy variables</span></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_impute_knn</span>(wbc) <span class="sc">%&gt;%</span> </span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_normalize</span>(<span class="fu">all_numeric_predictors</span>()) <span class="co"># normalize numeric features</span></span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>condition_metadata_res <span class="ot">=</span> <span class="fu">our_model</span>(model_data,<span class="st">"condition"</span>,</span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a>                                   faecalis_v_faecium_m_model_recipe)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>→ A | warning: ! There are new levels in `solid_tumor`: "FALSE".
               ℹ Consider using step_novel() (`?recipes::step_novel()`) \ before
                 `step_dummy()` to handle unseen values.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>There were issues with some computations   A: x1
There were issues with some computations   A: x1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">extract_roc</span>(condition_metadata_res,<span class="st">"condition"</span>,<span class="st">".pred_faecalis"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="machine_learning_analysis_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>Figure 7.</strong> The model based on the metadata alone performs well. It actually performs better than our metabolomics/proteomics model.</p>
<p>What features are important?</p>
<div class="cell" data-fig.showtext="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>condition_metadata_res<span class="sc">$</span>model_summary_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="machine_learning_analysis_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>Figure 8.</strong> Here we can see that the model values amp_mic as the most important features.</p>
<p>For all intents and purposes, this is cheating because by the time you perform a MIC, you already know what microbe is underlying the infection.</p>
<p>The whole point of our study was to use methods that would allow you to perform a prediction faster (1-3 days in advance</p>
<p>What if we remove variables that allow you to predict microbe identity (and were captured several days after admittance) from the metadata model?</p>
<div class="cell" data-fig.showtext="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>faecalis_v_faecium_m_model_recipe2 <span class="ot">&lt;-</span> <span class="fu">recipe</span>(condition <span class="sc">~</span> ., <span class="at">data =</span> model_data) <span class="sc">%&gt;%</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(sample_id, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(death_during_admission, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span>  <span class="co"># id role tells the model not to actually incorporate it as a feature</span></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(condition, <span class="at">new_role =</span> <span class="st">"outcome"</span>) <span class="sc">%&gt;%</span><span class="co"># specify target </span></span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(pathogen_type,<span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(sensitive_to_vancomycin,<span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(amp_mic,<span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(vanc_mic,<span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_novel</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_dummy</span>(<span class="fu">all_nominal_predictors</span>()) <span class="sc">%&gt;%</span> <span class="co"># convert string to dummy variables</span></span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_impute_knn</span>(wbc) <span class="sc">%&gt;%</span> </span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_normalize</span>(<span class="fu">all_numeric_predictors</span>()) <span class="co"># normalize numeric features</span></span>
<span id="cb45-13"><a href="#cb45-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-14"><a href="#cb45-14" aria-hidden="true" tabindex="-1"></a>condition_metadata_res2 <span class="ot">=</span> <span class="fu">our_model</span>(model_data,<span class="st">"condition"</span>,</span>
<span id="cb45-15"><a href="#cb45-15" aria-hidden="true" tabindex="-1"></a>                                   faecalis_v_faecium_m_model_recipe2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>→ A | warning: ! There are new levels in `solid_tumor`: "FALSE".
               ℹ Consider using step_novel() (`?recipes::step_novel()`) \ before
                 `step_dummy()` to handle unseen values.</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">extract_roc</span>(condition_metadata_res2,<span class="st">"condition"</span>,<span class="st">".pred_faecalis"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="machine_learning_analysis_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>Figure 8.</strong> When we do this, we find that that model performs slightly worse than our metabolomics/proteomics model, but on the same scale.</p>
<div class="cell" data-fig.showtext="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>condition_metadata_res2<span class="sc">$</span>model_summary_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="machine_learning_analysis_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>Figure 9.</strong> When we look into what is driving this model, we see that whether or not the patients had a transplant is a strong predictor.</p>
<p>This is not novel, and highlights how if you are immunocompromised, you are more likely to get a E.faecium infection.</p>
<p>What if we take out transplant type from the model?</p>
<div class="cell" data-fig.showtext="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>faecalis_v_faecium_m_model_recipe3 <span class="ot">&lt;-</span> <span class="fu">recipe</span>(condition <span class="sc">~</span> ., <span class="at">data =</span> model_data) <span class="sc">%&gt;%</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(sample_id, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(death_during_admission, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(pathogen_type,<span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(sensitive_to_vancomycin,<span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(amp_mic,<span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(vanc_mic,<span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(transplant_type, <span class="at">new_role =</span> <span class="st">"id"</span>)<span class="sc">%&gt;%</span>  <span class="co"># id role tells the model not to actually incorporate it as a feature</span></span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(condition, <span class="at">new_role =</span> <span class="st">"outcome"</span>) <span class="sc">%&gt;%</span> <span class="co"># specify target </span></span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_novel</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_dummy</span>(<span class="fu">all_nominal_predictors</span>()) <span class="sc">%&gt;%</span> <span class="co"># convert string to dummy variables</span></span>
<span id="cb49-12"><a href="#cb49-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_impute_knn</span>(wbc) <span class="sc">%&gt;%</span> </span>
<span id="cb49-13"><a href="#cb49-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_normalize</span>(<span class="fu">all_numeric_predictors</span>()) <span class="co"># normalize numeric features</span></span>
<span id="cb49-14"><a href="#cb49-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-15"><a href="#cb49-15" aria-hidden="true" tabindex="-1"></a>condition_metadata_res3 <span class="ot">=</span> <span class="fu">our_model</span>(model_data,<span class="st">"condition"</span>,</span>
<span id="cb49-16"><a href="#cb49-16" aria-hidden="true" tabindex="-1"></a>                                   faecalis_v_faecium_m_model_recipe3)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>→ A | warning: ! There are new levels in `solid_tumor`: "FALSE".
               ℹ Consider using step_novel() (`?recipes::step_novel()`) \ before
                 `step_dummy()` to handle unseen values.</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">extract_roc</span>(condition_metadata_res3,<span class="st">"condition"</span>,<span class="st">".pred_faecalis"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="machine_learning_analysis_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>Figure 10.</strong> Here we can see once all of these variables are taken out, the metadata model is essentially worthless.</p>
<p>All this to say, we did not find any clinical variables that are novel and predictive.</p>
<div class="cell" data-fig.showtext="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>condition_metadata_res3<span class="sc">$</span>model_summary_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="machine_learning_analysis_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>Figure 11.</strong> Here are the features that are important for the worthless model.</p>
</section>
</section>
<section id="mortality-vs-survival" class="level2">
<h2 class="anchored" data-anchor-id="mortality-vs-survival">Mortality vs Survival</h2>
<p>Now lets see how well we can predict mortality and survival using the proteomics and metabolomics data.</p>
<div class="cell" data-fig.showtext="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Have to discard the healthy data here </span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>model_data <span class="ot">&lt;-</span> clinical_metadata_o <span class="sc">%&gt;%</span></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(all_t, <span class="at">by =</span> <span class="st">"sample_id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(condition <span class="sc">!=</span> <span class="st">"healthy"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">death_during_admission =</span> <span class="fu">as.factor</span>(death_during_admission))</span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-8"><a href="#cb53-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Recipe does all of the feature engineering so that we don't have to do anything else to the input data</span></span>
<span id="cb53-9"><a href="#cb53-9" aria-hidden="true" tabindex="-1"></a>mortality_vs_survival_recipe <span class="ot">&lt;-</span> <span class="fu">recipe</span>(death_during_admission <span class="sc">~</span> ., <span class="at">data =</span> model_data) <span class="sc">%&gt;%</span></span>
<span id="cb53-10"><a href="#cb53-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(sample_id, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb53-11"><a href="#cb53-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(infection_status, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb53-12"><a href="#cb53-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(death_during_admission, <span class="at">new_role =</span> <span class="st">"outcome"</span>) <span class="sc">%&gt;%</span>  <span class="co"># id role tells the model not to actually incorporate it as a feature</span></span>
<span id="cb53-13"><a href="#cb53-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(condition, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> <span class="co"># specify target </span></span>
<span id="cb53-14"><a href="#cb53-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_dummy</span>(<span class="fu">all_nominal_predictors</span>()) <span class="sc">%&gt;%</span> <span class="co"># convert string to dummy variables</span></span>
<span id="cb53-15"><a href="#cb53-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_normalize</span>(<span class="fu">all_numeric_predictors</span>()) <span class="co"># normalize numeric features</span></span>
<span id="cb53-16"><a href="#cb53-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-17"><a href="#cb53-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-18"><a href="#cb53-18" aria-hidden="true" tabindex="-1"></a>mortality_vs_survival_res <span class="ot">=</span> <span class="fu">our_model</span>(model_data,<span class="st">"death_during_admission"</span>,mortality_vs_survival_recipe)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-fig.showtext="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">extract_roc</span>(mortality_vs_survival_res,<span class="st">"death_during_admission"</span>,<span class="st">".pred_FALSE"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="machine_learning_analysis_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>Figure 12.</strong> Here is a ROC curve of our model. It looks like it is reasonably accurate, but note that there is a big class imbalance here. A simple model where no patients die would get it right 76% of the time (57/75). Our fancy model is only slightly slightly improved on this. We are really hindered by our ability to only test a total of 3 true positives (due to the small sample size).</p>
<div class="cell" data-fig.showtext="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>mortality_vs_survival_res<span class="sc">$</span>confusion_matrix</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>          Reference
Prediction FALSE TRUE
     FALSE    11    1
     TRUE      2    2</code></pre>
</div>
</div>
<p><strong>Table 1.</strong> Here a confusion matrix highlights this class imbalance.</p>
<div class="cell" data-fig.showtext="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>mortality_vs_survival_res<span class="sc">$</span>model_summary_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="machine_learning_analysis_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>Figure 13.</strong> Here we can see the features important for the model. Note the size of the intercept.</p>
<section id="metadata-alone-model-1" class="level3">
<h3 class="anchored" data-anchor-id="metadata-alone-model-1">Metadata alone model</h3>
<p>I am curious how well a model made with just the metadata would perform.</p>
<div class="cell" data-fig.showtext="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>model_data <span class="ot">&lt;-</span> clinical_metadata <span class="sc">%&gt;%</span> </span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(condition <span class="sc">!=</span> <span class="st">"healthy"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># antimicrobial testing was not perfomred for this patient</span></span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sample_id <span class="sc">!=</span> <span class="st">"S18"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>  <span class="co">#getting rid of uninformative columns. Either very little variance, or too many levels to be useful </span></span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>pathogen,<span class="sc">-</span>patient_id,<span class="sc">-</span>pathogen_source,<span class="sc">-</span>infection_status,</span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>         <span class="sc">-</span>bacteremia_recurrence,<span class="sc">-</span>race,<span class="sc">-</span>death_at_one_year,</span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a>         <span class="sc">-</span>peripheral_vascular_disease,<span class="sc">-</span>dementia,<span class="sc">-</span>hemiplegia,</span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a>         <span class="sc">-</span>highly_active_antiretroviral_therapy,</span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a>         <span class="sc">-</span>source) <span class="sc">%&gt;%</span> </span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">transplant_type =</span> <span class="fu">case_when</span>(transplant_type <span class="sc">==</span> <span class="st">"none"</span> <span class="sc">~</span> <span class="st">"none"</span>,</span>
<span id="cb58-12"><a href="#cb58-12" aria-hidden="true" tabindex="-1"></a>                                     <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"transplant"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb58-13"><a href="#cb58-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select_if</span>(<span class="sc">~</span> <span class="sc">!</span><span class="fu">any</span>(<span class="fu">is.na</span>(.))) <span class="sc">%&gt;%</span> </span>
<span id="cb58-14"><a href="#cb58-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>antibiotic_therapy_before_48_hrs,<span class="sc">-</span>antibiotic_therapy_after_48_hrs) <span class="sc">%&gt;%</span> </span>
<span id="cb58-15"><a href="#cb58-15" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(history_of_myocardial_infarction, </span>
<span id="cb58-16"><a href="#cb58-16" aria-hidden="true" tabindex="-1"></a>                   congestive_heart_failure, </span>
<span id="cb58-17"><a href="#cb58-17" aria-hidden="true" tabindex="-1"></a>                   history_of_stroke_or_transient_ischemic_attack, </span>
<span id="cb58-18"><a href="#cb58-18" aria-hidden="true" tabindex="-1"></a>                   smoking, copd, peptic_ulcer_disaese, </span>
<span id="cb58-19"><a href="#cb58-19" aria-hidden="true" tabindex="-1"></a>                   mod_to_severe_chromic_kidney_disease, </span>
<span id="cb58-20"><a href="#cb58-20" aria-hidden="true" tabindex="-1"></a>                   hodgkin_disease, leukemia_or_lymphoma, </span>
<span id="cb58-21"><a href="#cb58-21" aria-hidden="true" tabindex="-1"></a>                   icu_admission, mechanical_ventilation, </span>
<span id="cb58-22"><a href="#cb58-22" aria-hidden="true" tabindex="-1"></a>                    polymicrobial_bacteremia, </span>
<span id="cb58-23"><a href="#cb58-23" aria-hidden="true" tabindex="-1"></a>                   thrombocytopenia_plt_50, hypotension,sensitive_to_vancomycin,</span>
<span id="cb58-24"><a href="#cb58-24" aria-hidden="true" tabindex="-1"></a>                   death_during_admission), </span>
<span id="cb58-25"><a href="#cb58-25" aria-hidden="true" tabindex="-1"></a>                as.factor)) <span class="sc">%&gt;%</span> </span>
<span id="cb58-26"><a href="#cb58-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">wbc =</span> <span class="fu">as.numeric</span>(wbc))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `wbc = as.numeric(wbc)`.
Caused by warning:
! NAs introduced by coercion</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>mort_m_model_recipe <span class="ot">&lt;-</span> <span class="fu">recipe</span>(death_during_admission <span class="sc">~</span> ., <span class="at">data =</span> model_data) <span class="sc">%&gt;%</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(sample_id, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(death_during_admission, <span class="at">new_role =</span> <span class="st">"outcome"</span>) <span class="sc">%&gt;%</span>  <span class="co"># id role tells the model not to actually incorporate it as a feature</span></span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>  <span class="co">#update_role(condition, new_role = "outcome") %&gt;% # specify target </span></span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_dummy</span>(<span class="fu">all_nominal_predictors</span>()) <span class="sc">%&gt;%</span> <span class="co"># convert string to dummy variables</span></span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_impute_knn</span>(wbc) <span class="sc">%&gt;%</span> </span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_normalize</span>(<span class="fu">all_numeric_predictors</span>()) <span class="co"># normalize numeric features</span></span>
<span id="cb60-8"><a href="#cb60-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-9"><a href="#cb60-9" aria-hidden="true" tabindex="-1"></a>mort_metadata_res <span class="ot">=</span> <span class="fu">our_model</span>(model_data,<span class="st">"death_during_admission"</span>,</span>
<span id="cb60-10"><a href="#cb60-10" aria-hidden="true" tabindex="-1"></a>                              mort_m_model_recipe)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>→ A | warning: ! There are new levels in `solid_tumor`: "FALSE".
               ℹ Consider using step_novel() (`?recipes::step_novel()`) \ before
                 `step_dummy()` to handle unseen values.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>There were issues with some computations   A: x1
There were issues with some computations   A: x1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">extract_roc</span>(mort_metadata_res,<span class="st">"death_during_admission"</span>,<span class="st">".pred_FALSE"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="machine_learning_analysis_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>Figure 14.</strong> Here we see that the clinical metadata does better job of predicting mortality than oru</p>
<p>What is going on here?</p>
<div class="cell" data-fig.showtext="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>mort_metadata_res<span class="sc">$</span>model_summary_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="machine_learning_analysis_files/figure-html/unnamed-chunk-26-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>**Figure 15.* Turns out if you get admitted to ICU, that predicts mortality pretty well (obviously). Once again, this is kind of cheating, ICU admission is a feature that is only apparent after a patient is at the hospital for some time.</p>
<p>What if we remove this variable from the model?</p>
<div class="cell" data-fig.showtext="false">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>mort_m_model_recipe <span class="ot">&lt;-</span> <span class="fu">recipe</span>(death_during_admission <span class="sc">~</span> ., <span class="at">data =</span> model_data) <span class="sc">%&gt;%</span></span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(sample_id, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(death_during_admission, <span class="at">new_role =</span> <span class="st">"outcome"</span>) <span class="sc">%&gt;%</span>  <span class="co"># id role tells the model not to actually incorporate it as a feature</span></span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(icu_admission, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> <span class="co"># specify target </span></span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_dummy</span>(<span class="fu">all_nominal_predictors</span>()) <span class="sc">%&gt;%</span> <span class="co"># convert string to dummy variables</span></span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_impute_knn</span>(wbc) <span class="sc">%&gt;%</span> </span>
<span id="cb66-7"><a href="#cb66-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_normalize</span>(<span class="fu">all_numeric_predictors</span>()) <span class="co"># normalize numeric features</span></span>
<span id="cb66-8"><a href="#cb66-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-9"><a href="#cb66-9" aria-hidden="true" tabindex="-1"></a>mort_metadata_res <span class="ot">=</span> <span class="fu">our_model</span>(model_data,<span class="st">"death_during_admission"</span>,</span>
<span id="cb66-10"><a href="#cb66-10" aria-hidden="true" tabindex="-1"></a>                              mort_m_model_recipe)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>→ A | warning: ! There are new levels in `solid_tumor`: "FALSE".
               ℹ Consider using step_novel() (`?recipes::step_novel()`) \ before
                 `step_dummy()` to handle unseen values.</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">extract_roc</span>(mort_metadata_res,<span class="st">"death_during_admission"</span>,<span class="st">".pred_FALSE"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="machine_learning_analysis_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>Figure 16.</strong> If you take out the ICU admission, you lose all model performance. You actually would be better off just predicting no one dies.</p>
</section>
</section>
</section>
<section id="conclusion" class="level1">
<h1>Conclusion</h1>
<ol type="1">
<li><p>The results of the lasso regression models show very similar results/ interpretations to the other form of analysis that we conducted.</p></li>
<li><p>These analyses should adress Review 2 comments appropriately,</p></li>
</ol>
<!-- -->

</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb69" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "machine_learning_analysis"</span></span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span><span class="co"> </span></span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a><span class="co">    theme: flatly</span></span>
<span id="cb69-6"><a href="#cb69-6" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb69-7"><a href="#cb69-7" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb69-8"><a href="#cb69-8" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> visual</span></span>
<span id="cb69-9"><a href="#cb69-9" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb69-10"><a href="#cb69-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-11"><a href="#cb69-11" aria-hidden="true" tabindex="-1"></a><span class="fu"># Machine Learning Analysis </span></span>
<span id="cb69-12"><a href="#cb69-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-13"><a href="#cb69-13" aria-hidden="true" tabindex="-1"></a>Here we will be performing a machine learning analysis of our proteomics and </span>
<span id="cb69-14"><a href="#cb69-14" aria-hidden="true" tabindex="-1"></a>metabolomics data.</span>
<span id="cb69-15"><a href="#cb69-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-16"><a href="#cb69-16" aria-hidden="true" tabindex="-1"></a>We intend to evaluate several models focusing on 3 comparisons:</span>
<span id="cb69-17"><a href="#cb69-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-18"><a href="#cb69-18" aria-hidden="true" tabindex="-1"></a><span class="ss">1. </span>Healthy vs Infected patients</span>
<span id="cb69-19"><a href="#cb69-19" aria-hidden="true" tabindex="-1"></a><span class="ss">2. </span>E.faecalis vs E.faecium infected patients</span>
<span id="cb69-20"><a href="#cb69-20" aria-hidden="true" tabindex="-1"></a><span class="ss">3. </span>Mortality vs Surival. </span>
<span id="cb69-21"><a href="#cb69-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-22"><a href="#cb69-22" aria-hidden="true" tabindex="-1"></a>Our dataset has low n but high p, so we will apply a lasso regression in hopes </span>
<span id="cb69-23"><a href="#cb69-23" aria-hidden="true" tabindex="-1"></a>that this will prevent overfitting of the data. </span>
<span id="cb69-24"><a href="#cb69-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-25"><a href="#cb69-25" aria-hidden="true" tabindex="-1"></a>We will split our </span>
<span id="cb69-26"><a href="#cb69-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-27"><a href="#cb69-27" aria-hidden="true" tabindex="-1"></a><span class="fu">## Loading/Formatting Data</span></span>
<span id="cb69-28"><a href="#cb69-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-31"><a href="#cb69-31" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-32"><a href="#cb69-32" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load-libraries</span></span>
<span id="cb69-33"><a href="#cb69-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-34"><a href="#cb69-34" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb69-35"><a href="#cb69-35" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidymodels)</span>
<span id="cb69-36"><a href="#cb69-36" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(caret)</span>
<span id="cb69-37"><a href="#cb69-37" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(broom)</span>
<span id="cb69-38"><a href="#cb69-38" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb69-39"><a href="#cb69-39" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb69-40"><a href="#cb69-40" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(thematic)</span>
<span id="cb69-41"><a href="#cb69-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-42"><a href="#cb69-42" aria-hidden="true" tabindex="-1"></a>ggplot2<span class="sc">::</span><span class="fu">theme_set</span>(<span class="fu">theme_bw</span>())</span>
<span id="cb69-43"><a href="#cb69-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-44"><a href="#cb69-44" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-45"><a href="#cb69-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-48"><a href="#cb69-48" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-49"><a href="#cb69-49" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load-data</span></span>
<span id="cb69-50"><a href="#cb69-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-51"><a href="#cb69-51" aria-hidden="true" tabindex="-1"></a>clinical_metadata <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"clinical_metadata.csv"</span>) </span>
<span id="cb69-52"><a href="#cb69-52" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb69-53"><a href="#cb69-53" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Clinical metadata with only the relevant outcomes</span></span>
<span id="cb69-54"><a href="#cb69-54" aria-hidden="true" tabindex="-1"></a>clinical_metadata_o <span class="ot">=</span> clinical_metadata <span class="sc">%&gt;%</span> </span>
<span id="cb69-55"><a href="#cb69-55" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(sample_id, condition, infection_status,</span>
<span id="cb69-56"><a href="#cb69-56" aria-hidden="true" tabindex="-1"></a>           death_during_admission)</span>
<span id="cb69-57"><a href="#cb69-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-58"><a href="#cb69-58" aria-hidden="true" tabindex="-1"></a>proteomics_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">'proteomics_data_cleaned.csv'</span>)</span>
<span id="cb69-59"><a href="#cb69-59" aria-hidden="true" tabindex="-1"></a>metabolomics_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">'metabolomics_data_cleaned.csv'</span>)</span>
<span id="cb69-60"><a href="#cb69-60" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-61"><a href="#cb69-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-64"><a href="#cb69-64" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-65"><a href="#cb69-65" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: transpose-data</span></span>
<span id="cb69-66"><a href="#cb69-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-67"><a href="#cb69-67" aria-hidden="true" tabindex="-1"></a><span class="co"># Transposing proteomics</span></span>
<span id="cb69-68"><a href="#cb69-68" aria-hidden="true" tabindex="-1"></a>proteomics_t <span class="ot">&lt;-</span> proteomics_data <span class="sc">%&gt;%</span></span>
<span id="cb69-69"><a href="#cb69-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>orientation, <span class="sc">-</span>uniprotid, <span class="sc">-</span>locus) <span class="sc">%&gt;%</span> </span>
<span id="cb69-70"><a href="#cb69-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(.),<span class="at">names_to =</span> <span class="st">"sample_id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-71"><a href="#cb69-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> Gene, <span class="at">values_from =</span> value)</span>
<span id="cb69-72"><a href="#cb69-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-73"><a href="#cb69-73" aria-hidden="true" tabindex="-1"></a><span class="co"># Transposing metabolomics  </span></span>
<span id="cb69-74"><a href="#cb69-74" aria-hidden="true" tabindex="-1"></a>metabolomics_t <span class="ot">&lt;-</span> metabolomics_data <span class="sc">%&gt;%</span> </span>
<span id="cb69-75"><a href="#cb69-75" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(compound_name)) <span class="sc">%&gt;%</span> </span>
<span id="cb69-76"><a href="#cb69-76" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">row_id =</span> <span class="fu">paste0</span>(row_id,<span class="st">":"</span>,compound_name)) <span class="sc">%&gt;%</span> </span>
<span id="cb69-77"><a href="#cb69-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>compound_name) <span class="sc">%&gt;%</span> </span>
<span id="cb69-78"><a href="#cb69-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(.),<span class="at">names_to =</span> <span class="st">"sample_id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-79"><a href="#cb69-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> row_id, <span class="at">values_from =</span> value)</span>
<span id="cb69-80"><a href="#cb69-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-81"><a href="#cb69-81" aria-hidden="true" tabindex="-1"></a><span class="co"># Combining proteomics and metabolomics</span></span>
<span id="cb69-82"><a href="#cb69-82" aria-hidden="true" tabindex="-1"></a>all_t <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">inner_join</span>(proteomics_t,metabolomics_t,<span class="at">by =</span> <span class="st">"sample_id"</span>)</span>
<span id="cb69-83"><a href="#cb69-83" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-84"><a href="#cb69-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-85"><a href="#cb69-85" aria-hidden="true" tabindex="-1"></a><span class="fu">## Model Function</span></span>
<span id="cb69-86"><a href="#cb69-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-87"><a href="#cb69-87" aria-hidden="true" tabindex="-1"></a>Here we will define a function to do all the modeling steps. This way for each</span>
<span id="cb69-88"><a href="#cb69-88" aria-hidden="true" tabindex="-1"></a>comparison, we only need to provide the model data and model recipie without </span>
<span id="cb69-89"><a href="#cb69-89" aria-hidden="true" tabindex="-1"></a>repeating everything over and over. </span>
<span id="cb69-90"><a href="#cb69-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-93"><a href="#cb69-93" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-94"><a href="#cb69-94" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: defining functions</span></span>
<span id="cb69-95"><a href="#cb69-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-96"><a href="#cb69-96" aria-hidden="true" tabindex="-1"></a><span class="co"># Defining the model Max built as a function, so we can easily apply it to all </span></span>
<span id="cb69-97"><a href="#cb69-97" aria-hidden="true" tabindex="-1"></a><span class="co"># of our comparisons.</span></span>
<span id="cb69-98"><a href="#cb69-98" aria-hidden="true" tabindex="-1"></a>our_model <span class="ot">=</span> <span class="cf">function</span>(model_data,</span>
<span id="cb69-99"><a href="#cb69-99" aria-hidden="true" tabindex="-1"></a>                     outcome,</span>
<span id="cb69-100"><a href="#cb69-100" aria-hidden="true" tabindex="-1"></a>                     model_recipe){</span>
<span id="cb69-101"><a href="#cb69-101" aria-hidden="true" tabindex="-1"></a><span class="co"># Setting seed at 42 to make reproducible.  </span></span>
<span id="cb69-102"><a href="#cb69-102" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb69-103"><a href="#cb69-103" aria-hidden="true" tabindex="-1"></a><span class="co"># Create train/test split</span></span>
<span id="cb69-104"><a href="#cb69-104" aria-hidden="true" tabindex="-1"></a>train_test_split <span class="ot">&lt;-</span> <span class="fu">initial_split</span>(model_data,<span class="at">prop =</span> <span class="fl">0.8</span>,</span>
<span id="cb69-105"><a href="#cb69-105" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">strata =</span> outcome)</span>
<span id="cb69-106"><a href="#cb69-106" aria-hidden="true" tabindex="-1"></a>train_data <span class="ot">&lt;-</span> <span class="fu">training</span>(train_test_split)</span>
<span id="cb69-107"><a href="#cb69-107" aria-hidden="true" tabindex="-1"></a>test_data <span class="ot">&lt;-</span> <span class="fu">testing</span>(train_test_split)</span>
<span id="cb69-108"><a href="#cb69-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-109"><a href="#cb69-109" aria-hidden="true" tabindex="-1"></a><span class="co"># Setting a logistic regression. </span></span>
<span id="cb69-110"><a href="#cb69-110" aria-hidden="true" tabindex="-1"></a>log_reg <span class="ot">&lt;-</span> <span class="fu">logistic_reg</span>(</span>
<span id="cb69-111"><a href="#cb69-111" aria-hidden="true" tabindex="-1"></a>  <span class="at">mixture =</span> <span class="dv">1</span>, <span class="co"># Lasso regression</span></span>
<span id="cb69-112"><a href="#cb69-112" aria-hidden="true" tabindex="-1"></a>  <span class="at">penalty =</span> <span class="fu">tune</span>() <span class="co"># testing a few different penalty values</span></span>
<span id="cb69-113"><a href="#cb69-113" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-114"><a href="#cb69-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_engine</span>(<span class="st">"glmnet"</span>) </span>
<span id="cb69-115"><a href="#cb69-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-116"><a href="#cb69-116" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating a model workflow</span></span>
<span id="cb69-117"><a href="#cb69-117" aria-hidden="true" tabindex="-1"></a>model_workflow <span class="ot">&lt;-</span> <span class="fu">workflow</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb69-118"><a href="#cb69-118" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_recipe</span>(model_recipe) <span class="sc">%&gt;%</span> </span>
<span id="cb69-119"><a href="#cb69-119" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_model</span>(log_reg)</span>
<span id="cb69-120"><a href="#cb69-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-121"><a href="#cb69-121" aria-hidden="true" tabindex="-1"></a><span class="co"># 5 fold cross validation</span></span>
<span id="cb69-122"><a href="#cb69-122" aria-hidden="true" tabindex="-1"></a>cv_folds <span class="ot">&lt;-</span> <span class="fu">vfold_cv</span>(train_data, <span class="at">v =</span> <span class="dv">5</span>)</span>
<span id="cb69-123"><a href="#cb69-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-124"><a href="#cb69-124" aria-hidden="true" tabindex="-1"></a><span class="co"># Tune the penalty parameter</span></span>
<span id="cb69-125"><a href="#cb69-125" aria-hidden="true" tabindex="-1"></a>tune_res <span class="ot">&lt;-</span> <span class="fu">tune_grid</span>(</span>
<span id="cb69-126"><a href="#cb69-126" aria-hidden="true" tabindex="-1"></a>  model_workflow,</span>
<span id="cb69-127"><a href="#cb69-127" aria-hidden="true" tabindex="-1"></a>  <span class="at">resamples =</span> cv_folds,</span>
<span id="cb69-128"><a href="#cb69-128" aria-hidden="true" tabindex="-1"></a>  <span class="at">grid =</span> <span class="dv">5</span>,     <span class="co"># Try 5 values of the penalty</span></span>
<span id="cb69-129"><a href="#cb69-129" aria-hidden="true" tabindex="-1"></a>  <span class="at">metrics =</span> <span class="fu">metric_set</span>(roc_auc, accuracy), <span class="co"># generate performance metrics for roc_auc &amp; accuracy</span></span>
<span id="cb69-130"><a href="#cb69-130" aria-hidden="true" tabindex="-1"></a>  <span class="at">control =</span> <span class="fu">control_grid</span>(<span class="at">save_pred =</span> <span class="cn">TRUE</span>)</span>
<span id="cb69-131"><a href="#cb69-131" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb69-132"><a href="#cb69-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-133"><a href="#cb69-133" aria-hidden="true" tabindex="-1"></a><span class="co"># Show performance metrics and get the best model</span></span>
<span id="cb69-134"><a href="#cb69-134" aria-hidden="true" tabindex="-1"></a>best_result <span class="ot">&lt;-</span> tune_res <span class="sc">%&gt;%</span></span>
<span id="cb69-135"><a href="#cb69-135" aria-hidden="true" tabindex="-1"></a>  <span class="fu">collect_metrics</span>() <span class="sc">%&gt;%</span></span>
<span id="cb69-136"><a href="#cb69-136" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(.metric <span class="sc">==</span> <span class="st">"roc_auc"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb69-137"><a href="#cb69-137" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(mean, <span class="at">n =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb69-138"><a href="#cb69-138" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(<span class="dv">1</span>)</span>
<span id="cb69-139"><a href="#cb69-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-140"><a href="#cb69-140" aria-hidden="true" tabindex="-1"></a><span class="co"># Finalize the workflow using the model workflow and our best performing model</span></span>
<span id="cb69-141"><a href="#cb69-141" aria-hidden="true" tabindex="-1"></a>final_workflow <span class="ot">&lt;-</span> <span class="fu">finalize_workflow</span>(model_workflow, best_result)</span>
<span id="cb69-142"><a href="#cb69-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-143"><a href="#cb69-143" aria-hidden="true" tabindex="-1"></a><span class="co"># Fit the Final Model on the Training Data</span></span>
<span id="cb69-144"><a href="#cb69-144" aria-hidden="true" tabindex="-1"></a>fit_workflow <span class="ot">&lt;-</span> <span class="fu">fit</span>(final_workflow, train_data)</span>
<span id="cb69-145"><a href="#cb69-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-146"><a href="#cb69-146" aria-hidden="true" tabindex="-1"></a><span class="co"># Coefficients for features that seem to matter to the model, the rest have been set to 0 via lasso regression</span></span>
<span id="cb69-147"><a href="#cb69-147" aria-hidden="true" tabindex="-1"></a>model_summary <span class="ot">&lt;-</span> <span class="fu">tidy</span>(fit_workflow)</span>
<span id="cb69-148"><a href="#cb69-148" aria-hidden="true" tabindex="-1"></a>model_summary_out <span class="ot">=</span> model_summary <span class="sc">%&gt;%</span></span>
<span id="cb69-149"><a href="#cb69-149" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(estimate <span class="sc">!=</span> <span class="dv">0</span>)</span>
<span id="cb69-150"><a href="#cb69-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-151"><a href="#cb69-151" aria-hidden="true" tabindex="-1"></a> <span class="co"># Make predictions on the test data</span></span>
<span id="cb69-152"><a href="#cb69-152" aria-hidden="true" tabindex="-1"></a>  predictions <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit_workflow, test_data, <span class="at">type =</span> <span class="st">"prob"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb69-153"><a href="#cb69-153" aria-hidden="true" tabindex="-1"></a>    <span class="fu">bind_cols</span>(<span class="fu">predict</span>(fit_workflow, test_data)) <span class="sc">%&gt;%</span></span>
<span id="cb69-154"><a href="#cb69-154" aria-hidden="true" tabindex="-1"></a>    <span class="fu">bind_cols</span>(test_data)</span>
<span id="cb69-155"><a href="#cb69-155" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb69-156"><a href="#cb69-156" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Generate confusion matrix</span></span>
<span id="cb69-157"><a href="#cb69-157" aria-hidden="true" tabindex="-1"></a>  pred <span class="ot">&lt;-</span> predictions <span class="sc">%&gt;%</span> <span class="fu">select</span>(outcome) <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">as.factor</span>()</span>
<span id="cb69-158"><a href="#cb69-158" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> predictions <span class="sc">%&gt;%</span> <span class="fu">select</span>(.pred_class) <span class="sc">%&gt;%</span> <span class="fu">pull</span>() <span class="sc">%&gt;%</span> <span class="fu">as.factor</span>()</span>
<span id="cb69-159"><a href="#cb69-159" aria-hidden="true" tabindex="-1"></a>  confusion_matrix <span class="ot">&lt;-</span> <span class="fu">confusionMatrix</span>(pred, res)</span>
<span id="cb69-160"><a href="#cb69-160" aria-hidden="true" tabindex="-1"></a>  confusion_matrix_out <span class="ot">&lt;-</span> confusion_matrix<span class="sc">$</span>table</span>
<span id="cb69-161"><a href="#cb69-161" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb69-162"><a href="#cb69-162" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plotting coefficients (Lasso-selected features)</span></span>
<span id="cb69-163"><a href="#cb69-163" aria-hidden="true" tabindex="-1"></a>  p1 <span class="ot">&lt;-</span> model_summary_out <span class="sc">%&gt;%</span> </span>
<span id="cb69-164"><a href="#cb69-164" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="fu">reorder</span>(term, estimate), estimate)) <span class="sc">+</span></span>
<span id="cb69-165"><a href="#cb69-165" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb69-166"><a href="#cb69-166" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>()</span>
<span id="cb69-167"><a href="#cb69-167" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb69-168"><a href="#cb69-168" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Return relevant outputs</span></span>
<span id="cb69-169"><a href="#cb69-169" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb69-170"><a href="#cb69-170" aria-hidden="true" tabindex="-1"></a>    <span class="at">model_summary =</span> model_summary_out,</span>
<span id="cb69-171"><a href="#cb69-171" aria-hidden="true" tabindex="-1"></a>    <span class="at">test_results =</span> predictions,</span>
<span id="cb69-172"><a href="#cb69-172" aria-hidden="true" tabindex="-1"></a>    <span class="at">model_summary_plot =</span> p1,</span>
<span id="cb69-173"><a href="#cb69-173" aria-hidden="true" tabindex="-1"></a>    <span class="at">confusion_matrix =</span> confusion_matrix_out</span>
<span id="cb69-174"><a href="#cb69-174" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb69-175"><a href="#cb69-175" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(out)</span>
<span id="cb69-176"><a href="#cb69-176" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb69-177"><a href="#cb69-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-178"><a href="#cb69-178" aria-hidden="true" tabindex="-1"></a><span class="co"># Extracting the ROC curves from the output of our_model()</span></span>
<span id="cb69-179"><a href="#cb69-179" aria-hidden="true" tabindex="-1"></a>extract_roc <span class="ot">=</span> <span class="cf">function</span>(list,outcome,prediction_col){</span>
<span id="cb69-180"><a href="#cb69-180" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb69-181"><a href="#cb69-181" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Convert the outcome variable to a factor</span></span>
<span id="cb69-182"><a href="#cb69-182" aria-hidden="true" tabindex="-1"></a>  predictions <span class="ot">&lt;-</span> list<span class="sc">$</span>test_results <span class="sc">%&gt;%</span> </span>
<span id="cb69-183"><a href="#cb69-183" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="sc">!!</span><span class="at">outcome :=</span> <span class="fu">as.factor</span>(<span class="sc">!!</span><span class="fu">sym</span>(outcome)))</span>
<span id="cb69-184"><a href="#cb69-184" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb69-185"><a href="#cb69-185" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Generate ROC curve data</span></span>
<span id="cb69-186"><a href="#cb69-186" aria-hidden="true" tabindex="-1"></a>  roc_data <span class="ot">&lt;-</span> <span class="fu">roc_curve</span>(predictions, <span class="sc">!!</span><span class="fu">sym</span>(outcome), <span class="sc">!!</span><span class="fu">sym</span>(prediction_col))</span>
<span id="cb69-187"><a href="#cb69-187" aria-hidden="true" tabindex="-1"></a>  roc_auc <span class="ot">&lt;-</span> <span class="fu">roc_auc</span>(predictions, <span class="sc">!!</span><span class="fu">sym</span>(outcome), <span class="sc">!!</span><span class="fu">sym</span>(prediction_col))</span>
<span id="cb69-188"><a href="#cb69-188" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb69-189"><a href="#cb69-189" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot ROC curve</span></span>
<span id="cb69-190"><a href="#cb69-190" aria-hidden="true" tabindex="-1"></a>  p1 <span class="ot">&lt;-</span> <span class="fu">autoplot</span>(roc_data) <span class="sc">+</span></span>
<span id="cb69-191"><a href="#cb69-191" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(<span class="st">"AUC = "</span>, <span class="fu">round</span>(roc_auc<span class="sc">$</span>.estimate, <span class="dv">2</span>)))</span>
<span id="cb69-192"><a href="#cb69-192" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb69-193"><a href="#cb69-193" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(p1)</span>
<span id="cb69-194"><a href="#cb69-194" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb69-195"><a href="#cb69-195" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-196"><a href="#cb69-196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-197"><a href="#cb69-197" aria-hidden="true" tabindex="-1"></a><span class="fu">## Healthy vs Infected</span></span>
<span id="cb69-198"><a href="#cb69-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-199"><a href="#cb69-199" aria-hidden="true" tabindex="-1"></a>First, let's see how well our model is able to predict healthy vs infected. </span>
<span id="cb69-200"><a href="#cb69-200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-203"><a href="#cb69-203" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-204"><a href="#cb69-204" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: generate-model-data</span></span>
<span id="cb69-205"><a href="#cb69-205" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb69-206"><a href="#cb69-206" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: true</span></span>
<span id="cb69-207"><a href="#cb69-207" aria-hidden="true" tabindex="-1"></a>model_data <span class="ot">&lt;-</span> clinical_metadata_o <span class="sc">%&gt;%</span></span>
<span id="cb69-208"><a href="#cb69-208" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(all_t, <span class="at">by =</span> <span class="st">"sample_id"</span>) </span>
<span id="cb69-209"><a href="#cb69-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-210"><a href="#cb69-210" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-211"><a href="#cb69-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-212"><a href="#cb69-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-215"><a href="#cb69-215" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-216"><a href="#cb69-216" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: create-healthy-v-infected-model-recipe</span></span>
<span id="cb69-217"><a href="#cb69-217" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb69-218"><a href="#cb69-218" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb69-219"><a href="#cb69-219" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-220"><a href="#cb69-220" aria-hidden="true" tabindex="-1"></a><span class="co"># Recipe does all of the feature engineering so that we don't have to do anything else to the input data</span></span>
<span id="cb69-221"><a href="#cb69-221" aria-hidden="true" tabindex="-1"></a>h_v_i_model_recipe <span class="ot">&lt;-</span> <span class="fu">recipe</span>(infection_status <span class="sc">~</span> ., <span class="at">data =</span> model_data) <span class="sc">%&gt;%</span></span>
<span id="cb69-222"><a href="#cb69-222" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(sample_id, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-223"><a href="#cb69-223" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(condition, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-224"><a href="#cb69-224" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(death_during_admission, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span>  <span class="co"># id role tells the model not to actually incorporate it as a feature</span></span>
<span id="cb69-225"><a href="#cb69-225" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(infection_status, <span class="at">new_role =</span> <span class="st">"outcome"</span>) <span class="sc">%&gt;%</span> <span class="co"># specify target </span></span>
<span id="cb69-226"><a href="#cb69-226" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_dummy</span>(<span class="fu">all_nominal_predictors</span>()) <span class="sc">%&gt;%</span> <span class="co"># convert string to dummy variables</span></span>
<span id="cb69-227"><a href="#cb69-227" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_normalize</span>(<span class="fu">all_numeric_predictors</span>()) <span class="co"># normalize numeric features</span></span>
<span id="cb69-228"><a href="#cb69-228" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-229"><a href="#cb69-229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-230"><a href="#cb69-230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-231"><a href="#cb69-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-234"><a href="#cb69-234" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-235"><a href="#cb69-235" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: view ROC </span></span>
<span id="cb69-236"><a href="#cb69-236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-237"><a href="#cb69-237" aria-hidden="true" tabindex="-1"></a>h_v_i_res <span class="ot">&lt;-</span> <span class="fu">our_model</span>(model_data,<span class="st">"infection_status"</span>,h_v_i_model_recipe)</span>
<span id="cb69-238"><a href="#cb69-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-239"><a href="#cb69-239" aria-hidden="true" tabindex="-1"></a><span class="fu">extract_roc</span>(h_v_i_res,<span class="st">"infection_status"</span>,<span class="st">".pred_infected"</span>)</span>
<span id="cb69-240"><a href="#cb69-240" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-241"><a href="#cb69-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-242"><a href="#cb69-242" aria-hidden="true" tabindex="-1"></a>**Figure 1.** Here we see that the model performs perfectly. </span>
<span id="cb69-243"><a href="#cb69-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-244"><a href="#cb69-244" aria-hidden="true" tabindex="-1"></a>What features are important for this model?</span>
<span id="cb69-245"><a href="#cb69-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-248"><a href="#cb69-248" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-249"><a href="#cb69-249" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: view estimates </span></span>
<span id="cb69-250"><a href="#cb69-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-251"><a href="#cb69-251" aria-hidden="true" tabindex="-1"></a>h_v_i_res<span class="sc">$</span>model_summary_plot</span>
<span id="cb69-252"><a href="#cb69-252" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-253"><a href="#cb69-253" aria-hidden="true" tabindex="-1"></a>**Figure 2.** Here we see the features that are important for the model. Several of these features are likely artifacts of the way the samples were collected. Phtalic anhydride and phthalate are plasticizers. These are almost certainly due to the fact that these samples were collected in different types fo plastic tubes. This was a point the reviewers brought up. </span>
<span id="cb69-254"><a href="#cb69-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-255"><a href="#cb69-255" aria-hidden="true" tabindex="-1"></a>We should either run the model with these features removed, or entirely remove the metabolomics data from this comparison. </span>
<span id="cb69-256"><a href="#cb69-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-257"><a href="#cb69-257" aria-hidden="true" tabindex="-1"></a>Also, note the value of the intercept. This is a reasonable value as there is </span>
<span id="cb69-258"><a href="#cb69-258" aria-hidden="true" tabindex="-1"></a>some class imbalance here since we have 76 infected patients and only 29 </span>
<span id="cb69-259"><a href="#cb69-259" aria-hidden="true" tabindex="-1"></a>uninfected. </span>
<span id="cb69-260"><a href="#cb69-260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-263"><a href="#cb69-263" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-264"><a href="#cb69-264" aria-hidden="true" tabindex="-1"></a><span class="co"># Recipe does all of the feature engineering so that we don't have to do anything else to the input data</span></span>
<span id="cb69-265"><a href="#cb69-265" aria-hidden="true" tabindex="-1"></a>h_v_i_model_recipe2 <span class="ot">&lt;-</span> <span class="fu">recipe</span>(infection_status <span class="sc">~</span> ., <span class="at">data =</span> model_data) <span class="sc">%&gt;%</span></span>
<span id="cb69-266"><a href="#cb69-266" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(sample_id, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-267"><a href="#cb69-267" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(condition, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-268"><a href="#cb69-268" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(<span class="st">`</span><span class="at">RID_29112:bis(2-ethylhexyl) phthalate</span><span class="st">`</span> , <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-269"><a href="#cb69-269" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(<span class="st">`</span><span class="at">RID_22644:phthalic anhydride</span><span class="st">`</span> , <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-270"><a href="#cb69-270" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(death_during_admission, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span>  <span class="co"># id role tells the model not to actually incorporate it as a feature</span></span>
<span id="cb69-271"><a href="#cb69-271" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(infection_status, <span class="at">new_role =</span> <span class="st">"outcome"</span>) <span class="sc">%&gt;%</span> <span class="co"># specify target </span></span>
<span id="cb69-272"><a href="#cb69-272" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_dummy</span>(<span class="fu">all_nominal_predictors</span>()) <span class="sc">%&gt;%</span> <span class="co"># convert string to dummy variables</span></span>
<span id="cb69-273"><a href="#cb69-273" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_normalize</span>(<span class="fu">all_numeric_predictors</span>()) <span class="co"># normalize numeric features</span></span>
<span id="cb69-274"><a href="#cb69-274" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-275"><a href="#cb69-275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-278"><a href="#cb69-278" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-279"><a href="#cb69-279" aria-hidden="true" tabindex="-1"></a>h_v_i_res2 <span class="ot">&lt;-</span> <span class="fu">our_model</span>(model_data,<span class="st">"infection_status"</span>,h_v_i_model_recipe2)</span>
<span id="cb69-280"><a href="#cb69-280" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-281"><a href="#cb69-281" aria-hidden="true" tabindex="-1"></a><span class="fu">extract_roc</span>(h_v_i_res2,<span class="st">"infection_status"</span>,<span class="st">".pred_infected"</span>)</span>
<span id="cb69-282"><a href="#cb69-282" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-283"><a href="#cb69-283" aria-hidden="true" tabindex="-1"></a>**Figure 3.** Model still performs perfectly with problematic metabolites removed. </span>
<span id="cb69-284"><a href="#cb69-284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-287"><a href="#cb69-287" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-288"><a href="#cb69-288" aria-hidden="true" tabindex="-1"></a>h_v_i_res2<span class="sc">$</span>model_summary_plot</span>
<span id="cb69-289"><a href="#cb69-289" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-290"><a href="#cb69-290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-291"><a href="#cb69-291" aria-hidden="true" tabindex="-1"></a>**Figure 4.** Here we can see the features important to this new model.</span>
<span id="cb69-292"><a href="#cb69-292" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-293"><a href="#cb69-293" aria-hidden="true" tabindex="-1"></a><span class="fu">## *E. faecalis* vs *E. faecium*</span></span>
<span id="cb69-294"><a href="#cb69-294" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-295"><a href="#cb69-295" aria-hidden="true" tabindex="-1"></a>Now let's see how well the modeling performs for distinguishing *E.faecalis* from *E.faecium*</span>
<span id="cb69-296"><a href="#cb69-296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-299"><a href="#cb69-299" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-300"><a href="#cb69-300" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: faecalis v faecium model </span></span>
<span id="cb69-301"><a href="#cb69-301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-302"><a href="#cb69-302" aria-hidden="true" tabindex="-1"></a><span class="co"># Have to discard the healthy data here </span></span>
<span id="cb69-303"><a href="#cb69-303" aria-hidden="true" tabindex="-1"></a>model_data <span class="ot">&lt;-</span> clinical_metadata_o <span class="sc">%&gt;%</span></span>
<span id="cb69-304"><a href="#cb69-304" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(all_t, <span class="at">by =</span> <span class="st">"sample_id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-305"><a href="#cb69-305" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(condition <span class="sc">!=</span> <span class="st">"healthy"</span>)</span>
<span id="cb69-306"><a href="#cb69-306" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-307"><a href="#cb69-307" aria-hidden="true" tabindex="-1"></a>outcome <span class="ot">=</span> <span class="st">"infection_status"</span></span>
<span id="cb69-308"><a href="#cb69-308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-309"><a href="#cb69-309" aria-hidden="true" tabindex="-1"></a><span class="co"># Recipe does all of the feature engineering so that we don't have to do anything else to the input data</span></span>
<span id="cb69-310"><a href="#cb69-310" aria-hidden="true" tabindex="-1"></a>faecalis_v_faecium_recipe <span class="ot">&lt;-</span> <span class="fu">recipe</span>(condition <span class="sc">~</span> ., <span class="at">data =</span> model_data) <span class="sc">%&gt;%</span></span>
<span id="cb69-311"><a href="#cb69-311" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(sample_id, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-312"><a href="#cb69-312" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(infection_status, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-313"><a href="#cb69-313" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(death_during_admission, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span>  <span class="co"># id role tells the model not to actually incorporate it as a feature</span></span>
<span id="cb69-314"><a href="#cb69-314" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(condition, <span class="at">new_role =</span> <span class="st">"outcome"</span>) <span class="sc">%&gt;%</span> <span class="co"># specify target </span></span>
<span id="cb69-315"><a href="#cb69-315" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_dummy</span>(<span class="fu">all_nominal_predictors</span>()) <span class="sc">%&gt;%</span> <span class="co"># convert string to dummy variables</span></span>
<span id="cb69-316"><a href="#cb69-316" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_normalize</span>(<span class="fu">all_numeric_predictors</span>()) <span class="co"># normalize numeric features</span></span>
<span id="cb69-317"><a href="#cb69-317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-318"><a href="#cb69-318" aria-hidden="true" tabindex="-1"></a>faecalis_v_faecium_res <span class="ot">&lt;-</span> <span class="fu">our_model</span>(model_data,<span class="st">"condition"</span>,faecalis_v_faecium_recipe)</span>
<span id="cb69-319"><a href="#cb69-319" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-320"><a href="#cb69-320" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-323"><a href="#cb69-323" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-324"><a href="#cb69-324" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: model ROCs</span></span>
<span id="cb69-325"><a href="#cb69-325" aria-hidden="true" tabindex="-1"></a><span class="fu">extract_roc</span>(faecalis_v_faecium_res,<span class="st">"condition"</span>,<span class="st">".pred_faecalis"</span>)</span>
<span id="cb69-326"><a href="#cb69-326" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-327"><a href="#cb69-327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-328"><a href="#cb69-328" aria-hidden="true" tabindex="-1"></a>**Figure 5.** Here we can see the ROC curve for the E.facalis vs E.faecium prediction. The model looks like it is performing okay here. This isn't good enough to be clinically useful really. </span>
<span id="cb69-329"><a href="#cb69-329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-330"><a href="#cb69-330" aria-hidden="true" tabindex="-1"></a>Note that a simple model where you assume everyone has E. faecalis</span>
<span id="cb69-331"><a href="#cb69-331" aria-hidden="true" tabindex="-1"></a>gets it right 57% of the time since there are 43 patients with faecalis and only </span>
<span id="cb69-332"><a href="#cb69-332" aria-hidden="true" tabindex="-1"></a>32 with faecium.</span>
<span id="cb69-333"><a href="#cb69-333" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-336"><a href="#cb69-336" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-337"><a href="#cb69-337" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: model estimates</span></span>
<span id="cb69-338"><a href="#cb69-338" aria-hidden="true" tabindex="-1"></a><span class="co">#| </span></span>
<span id="cb69-339"><a href="#cb69-339" aria-hidden="true" tabindex="-1"></a>faecalis_v_faecium_res<span class="sc">$</span>model_summary_plot</span>
<span id="cb69-340"><a href="#cb69-340" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-341"><a href="#cb69-341" aria-hidden="true" tabindex="-1"></a>**Figure 6.** Here we can see the features that are most important to the model. Looks very similar to the previous analysis. Several of these proteins and metabolites are immune related. </span>
<span id="cb69-342"><a href="#cb69-342" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-343"><a href="#cb69-343" aria-hidden="true" tabindex="-1"></a>Signal we are picking up here is probably that people with faecium infections are more likely to be immmunocompromised (this is already established in the literature.)</span>
<span id="cb69-344"><a href="#cb69-344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-345"><a href="#cb69-345" aria-hidden="true" tabindex="-1"></a><span class="fu">### Metadata alone model</span></span>
<span id="cb69-346"><a href="#cb69-346" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-347"><a href="#cb69-347" aria-hidden="true" tabindex="-1"></a>I am curious how well a model made with just the metadata would perform. </span>
<span id="cb69-348"><a href="#cb69-348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-349"><a href="#cb69-349" aria-hidden="true" tabindex="-1"></a>I would expect the MIC testing to do a good job of predicting faecium, as facieum isolates are more likely to be ampicillin and vancomycin resistant. </span>
<span id="cb69-350"><a href="#cb69-350" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-353"><a href="#cb69-353" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-354"><a href="#cb69-354" aria-hidden="true" tabindex="-1"></a>model_data <span class="ot">&lt;-</span> clinical_metadata <span class="sc">%&gt;%</span> </span>
<span id="cb69-355"><a href="#cb69-355" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(condition <span class="sc">!=</span> <span class="st">"healthy"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-356"><a href="#cb69-356" aria-hidden="true" tabindex="-1"></a>  <span class="co"># antimicrobial testing was not perfomred for this patient</span></span>
<span id="cb69-357"><a href="#cb69-357" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sample_id <span class="sc">!=</span> <span class="st">"S18"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-358"><a href="#cb69-358" aria-hidden="true" tabindex="-1"></a>  <span class="co">#getting rid of uninformative columns. Either very little variance, or too many levels to be useful </span></span>
<span id="cb69-359"><a href="#cb69-359" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>pathogen,<span class="sc">-</span>patient_id,<span class="sc">-</span>pathogen_source,<span class="sc">-</span>infection_status,</span>
<span id="cb69-360"><a href="#cb69-360" aria-hidden="true" tabindex="-1"></a>         <span class="sc">-</span>bacteremia_recurrence,<span class="sc">-</span>race,<span class="sc">-</span>death_at_one_year,</span>
<span id="cb69-361"><a href="#cb69-361" aria-hidden="true" tabindex="-1"></a>         <span class="sc">-</span>peripheral_vascular_disease,<span class="sc">-</span>dementia,<span class="sc">-</span>hemiplegia,</span>
<span id="cb69-362"><a href="#cb69-362" aria-hidden="true" tabindex="-1"></a>         <span class="sc">-</span>highly_active_antiretroviral_therapy,</span>
<span id="cb69-363"><a href="#cb69-363" aria-hidden="true" tabindex="-1"></a>         <span class="sc">-</span>source) <span class="sc">%&gt;%</span> </span>
<span id="cb69-364"><a href="#cb69-364" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">transplant_type =</span> <span class="fu">case_when</span>(transplant_type <span class="sc">==</span> <span class="st">"none"</span> <span class="sc">~</span> <span class="st">"none"</span>,</span>
<span id="cb69-365"><a href="#cb69-365" aria-hidden="true" tabindex="-1"></a>                                     <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"transplant"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb69-366"><a href="#cb69-366" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select_if</span>(<span class="sc">~</span> <span class="sc">!</span><span class="fu">any</span>(<span class="fu">is.na</span>(.))) <span class="sc">%&gt;%</span> </span>
<span id="cb69-367"><a href="#cb69-367" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>antibiotic_therapy_before_48_hrs,<span class="sc">-</span>antibiotic_therapy_after_48_hrs) <span class="sc">%&gt;%</span> </span>
<span id="cb69-368"><a href="#cb69-368" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(history_of_myocardial_infarction, </span>
<span id="cb69-369"><a href="#cb69-369" aria-hidden="true" tabindex="-1"></a>                   congestive_heart_failure, </span>
<span id="cb69-370"><a href="#cb69-370" aria-hidden="true" tabindex="-1"></a>                   history_of_stroke_or_transient_ischemic_attack, </span>
<span id="cb69-371"><a href="#cb69-371" aria-hidden="true" tabindex="-1"></a>                   smoking, copd, peptic_ulcer_disaese, </span>
<span id="cb69-372"><a href="#cb69-372" aria-hidden="true" tabindex="-1"></a>                   mod_to_severe_chromic_kidney_disease, </span>
<span id="cb69-373"><a href="#cb69-373" aria-hidden="true" tabindex="-1"></a>                   hodgkin_disease, leukemia_or_lymphoma, </span>
<span id="cb69-374"><a href="#cb69-374" aria-hidden="true" tabindex="-1"></a>                   icu_admission, mechanical_ventilation, </span>
<span id="cb69-375"><a href="#cb69-375" aria-hidden="true" tabindex="-1"></a>                    polymicrobial_bacteremia, </span>
<span id="cb69-376"><a href="#cb69-376" aria-hidden="true" tabindex="-1"></a>                   thrombocytopenia_plt_50, hypotension,sensitive_to_vancomycin), </span>
<span id="cb69-377"><a href="#cb69-377" aria-hidden="true" tabindex="-1"></a>                as.factor)) <span class="sc">%&gt;%</span> </span>
<span id="cb69-378"><a href="#cb69-378" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">wbc =</span> <span class="fu">as.numeric</span>(wbc))</span>
<span id="cb69-379"><a href="#cb69-379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-380"><a href="#cb69-380" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-381"><a href="#cb69-381" aria-hidden="true" tabindex="-1"></a>faecalis_v_faecium_m_model_recipe <span class="ot">&lt;-</span> <span class="fu">recipe</span>(condition <span class="sc">~</span> ., <span class="at">data =</span> model_data) <span class="sc">%&gt;%</span></span>
<span id="cb69-382"><a href="#cb69-382" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(sample_id, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-383"><a href="#cb69-383" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(death_during_admission, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span>  <span class="co"># id role tells the model not to actually incorporate it as a feature</span></span>
<span id="cb69-384"><a href="#cb69-384" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(condition, <span class="at">new_role =</span> <span class="st">"outcome"</span>) <span class="sc">%&gt;%</span> <span class="co"># specify target </span></span>
<span id="cb69-385"><a href="#cb69-385" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_novel</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb69-386"><a href="#cb69-386" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_dummy</span>(<span class="fu">all_nominal_predictors</span>()) <span class="sc">%&gt;%</span> <span class="co"># convert string to dummy variables</span></span>
<span id="cb69-387"><a href="#cb69-387" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_impute_knn</span>(wbc) <span class="sc">%&gt;%</span> </span>
<span id="cb69-388"><a href="#cb69-388" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_normalize</span>(<span class="fu">all_numeric_predictors</span>()) <span class="co"># normalize numeric features</span></span>
<span id="cb69-389"><a href="#cb69-389" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-390"><a href="#cb69-390" aria-hidden="true" tabindex="-1"></a>condition_metadata_res <span class="ot">=</span> <span class="fu">our_model</span>(model_data,<span class="st">"condition"</span>,</span>
<span id="cb69-391"><a href="#cb69-391" aria-hidden="true" tabindex="-1"></a>                                   faecalis_v_faecium_m_model_recipe)</span>
<span id="cb69-392"><a href="#cb69-392" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-393"><a href="#cb69-393" aria-hidden="true" tabindex="-1"></a><span class="fu">extract_roc</span>(condition_metadata_res,<span class="st">"condition"</span>,<span class="st">".pred_faecalis"</span>)</span>
<span id="cb69-394"><a href="#cb69-394" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-395"><a href="#cb69-395" aria-hidden="true" tabindex="-1"></a>**Figure 7.** The model based on the metadata alone performs well. It actually performs better than our metabolomics/proteomics model.</span>
<span id="cb69-396"><a href="#cb69-396" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-397"><a href="#cb69-397" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-398"><a href="#cb69-398" aria-hidden="true" tabindex="-1"></a>What features are important?</span>
<span id="cb69-399"><a href="#cb69-399" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-402"><a href="#cb69-402" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-403"><a href="#cb69-403" aria-hidden="true" tabindex="-1"></a>condition_metadata_res<span class="sc">$</span>model_summary_plot</span>
<span id="cb69-404"><a href="#cb69-404" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-405"><a href="#cb69-405" aria-hidden="true" tabindex="-1"></a>**Figure 8.** Here we can see that the model values amp_mic as the most important features. </span>
<span id="cb69-406"><a href="#cb69-406" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-407"><a href="#cb69-407" aria-hidden="true" tabindex="-1"></a>For all intents and purposes, this is cheating because by the time you perform a MIC, you already know what microbe is underlying the infection. </span>
<span id="cb69-408"><a href="#cb69-408" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-409"><a href="#cb69-409" aria-hidden="true" tabindex="-1"></a>The whole point of our study was to use methods that would allow you to perform a prediction faster (1-3 days in advance</span>
<span id="cb69-410"><a href="#cb69-410" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-411"><a href="#cb69-411" aria-hidden="true" tabindex="-1"></a>What if we remove variables that allow you to predict microbe identity (and were captured several days after admittance) from the metadata model?</span>
<span id="cb69-412"><a href="#cb69-412" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-415"><a href="#cb69-415" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-416"><a href="#cb69-416" aria-hidden="true" tabindex="-1"></a>faecalis_v_faecium_m_model_recipe2 <span class="ot">&lt;-</span> <span class="fu">recipe</span>(condition <span class="sc">~</span> ., <span class="at">data =</span> model_data) <span class="sc">%&gt;%</span></span>
<span id="cb69-417"><a href="#cb69-417" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(sample_id, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-418"><a href="#cb69-418" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(death_during_admission, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span>  <span class="co"># id role tells the model not to actually incorporate it as a feature</span></span>
<span id="cb69-419"><a href="#cb69-419" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(condition, <span class="at">new_role =</span> <span class="st">"outcome"</span>) <span class="sc">%&gt;%</span><span class="co"># specify target </span></span>
<span id="cb69-420"><a href="#cb69-420" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(pathogen_type,<span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-421"><a href="#cb69-421" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(sensitive_to_vancomycin,<span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-422"><a href="#cb69-422" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(amp_mic,<span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-423"><a href="#cb69-423" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(vanc_mic,<span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-424"><a href="#cb69-424" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_novel</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb69-425"><a href="#cb69-425" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_dummy</span>(<span class="fu">all_nominal_predictors</span>()) <span class="sc">%&gt;%</span> <span class="co"># convert string to dummy variables</span></span>
<span id="cb69-426"><a href="#cb69-426" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_impute_knn</span>(wbc) <span class="sc">%&gt;%</span> </span>
<span id="cb69-427"><a href="#cb69-427" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_normalize</span>(<span class="fu">all_numeric_predictors</span>()) <span class="co"># normalize numeric features</span></span>
<span id="cb69-428"><a href="#cb69-428" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-429"><a href="#cb69-429" aria-hidden="true" tabindex="-1"></a>condition_metadata_res2 <span class="ot">=</span> <span class="fu">our_model</span>(model_data,<span class="st">"condition"</span>,</span>
<span id="cb69-430"><a href="#cb69-430" aria-hidden="true" tabindex="-1"></a>                                   faecalis_v_faecium_m_model_recipe2)</span>
<span id="cb69-431"><a href="#cb69-431" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-432"><a href="#cb69-432" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-433"><a href="#cb69-433" aria-hidden="true" tabindex="-1"></a><span class="fu">extract_roc</span>(condition_metadata_res2,<span class="st">"condition"</span>,<span class="st">".pred_faecalis"</span>)</span>
<span id="cb69-434"><a href="#cb69-434" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-435"><a href="#cb69-435" aria-hidden="true" tabindex="-1"></a>**Figure 8.** When we do this, we find that that model performs slightly worse than our metabolomics/proteomics model, but on the same scale.</span>
<span id="cb69-436"><a href="#cb69-436" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-439"><a href="#cb69-439" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-440"><a href="#cb69-440" aria-hidden="true" tabindex="-1"></a>condition_metadata_res2<span class="sc">$</span>model_summary_plot</span>
<span id="cb69-441"><a href="#cb69-441" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-442"><a href="#cb69-442" aria-hidden="true" tabindex="-1"></a>**Figure 9.** When we look into what is driving this model, we see that whether or not the patients had a transplant is a strong predictor. </span>
<span id="cb69-443"><a href="#cb69-443" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-444"><a href="#cb69-444" aria-hidden="true" tabindex="-1"></a>This is not novel, and highlights how if you are immunocompromised, you are more likely to get a E.faecium infection. </span>
<span id="cb69-445"><a href="#cb69-445" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-446"><a href="#cb69-446" aria-hidden="true" tabindex="-1"></a>What if we take out transplant type from the model?</span>
<span id="cb69-449"><a href="#cb69-449" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-450"><a href="#cb69-450" aria-hidden="true" tabindex="-1"></a>faecalis_v_faecium_m_model_recipe3 <span class="ot">&lt;-</span> <span class="fu">recipe</span>(condition <span class="sc">~</span> ., <span class="at">data =</span> model_data) <span class="sc">%&gt;%</span></span>
<span id="cb69-451"><a href="#cb69-451" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(sample_id, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-452"><a href="#cb69-452" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(death_during_admission, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-453"><a href="#cb69-453" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(pathogen_type,<span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-454"><a href="#cb69-454" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(sensitive_to_vancomycin,<span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-455"><a href="#cb69-455" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(amp_mic,<span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-456"><a href="#cb69-456" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(vanc_mic,<span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-457"><a href="#cb69-457" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(transplant_type, <span class="at">new_role =</span> <span class="st">"id"</span>)<span class="sc">%&gt;%</span>  <span class="co"># id role tells the model not to actually incorporate it as a feature</span></span>
<span id="cb69-458"><a href="#cb69-458" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(condition, <span class="at">new_role =</span> <span class="st">"outcome"</span>) <span class="sc">%&gt;%</span> <span class="co"># specify target </span></span>
<span id="cb69-459"><a href="#cb69-459" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_novel</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb69-460"><a href="#cb69-460" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_dummy</span>(<span class="fu">all_nominal_predictors</span>()) <span class="sc">%&gt;%</span> <span class="co"># convert string to dummy variables</span></span>
<span id="cb69-461"><a href="#cb69-461" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_impute_knn</span>(wbc) <span class="sc">%&gt;%</span> </span>
<span id="cb69-462"><a href="#cb69-462" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_normalize</span>(<span class="fu">all_numeric_predictors</span>()) <span class="co"># normalize numeric features</span></span>
<span id="cb69-463"><a href="#cb69-463" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-464"><a href="#cb69-464" aria-hidden="true" tabindex="-1"></a>condition_metadata_res3 <span class="ot">=</span> <span class="fu">our_model</span>(model_data,<span class="st">"condition"</span>,</span>
<span id="cb69-465"><a href="#cb69-465" aria-hidden="true" tabindex="-1"></a>                                   faecalis_v_faecium_m_model_recipe3)</span>
<span id="cb69-466"><a href="#cb69-466" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-467"><a href="#cb69-467" aria-hidden="true" tabindex="-1"></a><span class="fu">extract_roc</span>(condition_metadata_res3,<span class="st">"condition"</span>,<span class="st">".pred_faecalis"</span>)</span>
<span id="cb69-468"><a href="#cb69-468" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-469"><a href="#cb69-469" aria-hidden="true" tabindex="-1"></a>**Figure 10.** Here we can see once all of these variables are taken out, the metadata model is essentially worthless. </span>
<span id="cb69-470"><a href="#cb69-470" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-471"><a href="#cb69-471" aria-hidden="true" tabindex="-1"></a>All this to say, we did not find any clinical variables that are novel and </span>
<span id="cb69-472"><a href="#cb69-472" aria-hidden="true" tabindex="-1"></a>predictive. </span>
<span id="cb69-473"><a href="#cb69-473" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-476"><a href="#cb69-476" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-477"><a href="#cb69-477" aria-hidden="true" tabindex="-1"></a>condition_metadata_res3<span class="sc">$</span>model_summary_plot</span>
<span id="cb69-478"><a href="#cb69-478" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-479"><a href="#cb69-479" aria-hidden="true" tabindex="-1"></a>**Figure 11.** Here are the features that are important for the worthless model. </span>
<span id="cb69-480"><a href="#cb69-480" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-481"><a href="#cb69-481" aria-hidden="true" tabindex="-1"></a><span class="fu">## Mortality vs Survival</span></span>
<span id="cb69-482"><a href="#cb69-482" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-483"><a href="#cb69-483" aria-hidden="true" tabindex="-1"></a>Now lets see how well we can predict mortality and survival using the proteomics and metabolomics data. </span>
<span id="cb69-484"><a href="#cb69-484" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-487"><a href="#cb69-487" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-488"><a href="#cb69-488" aria-hidden="true" tabindex="-1"></a><span class="co"># Have to discard the healthy data here </span></span>
<span id="cb69-489"><a href="#cb69-489" aria-hidden="true" tabindex="-1"></a>model_data <span class="ot">&lt;-</span> clinical_metadata_o <span class="sc">%&gt;%</span></span>
<span id="cb69-490"><a href="#cb69-490" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(all_t, <span class="at">by =</span> <span class="st">"sample_id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-491"><a href="#cb69-491" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(condition <span class="sc">!=</span> <span class="st">"healthy"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-492"><a href="#cb69-492" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">death_during_admission =</span> <span class="fu">as.factor</span>(death_during_admission))</span>
<span id="cb69-493"><a href="#cb69-493" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-494"><a href="#cb69-494" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-495"><a href="#cb69-495" aria-hidden="true" tabindex="-1"></a><span class="co"># Recipe does all of the feature engineering so that we don't have to do anything else to the input data</span></span>
<span id="cb69-496"><a href="#cb69-496" aria-hidden="true" tabindex="-1"></a>mortality_vs_survival_recipe <span class="ot">&lt;-</span> <span class="fu">recipe</span>(death_during_admission <span class="sc">~</span> ., <span class="at">data =</span> model_data) <span class="sc">%&gt;%</span></span>
<span id="cb69-497"><a href="#cb69-497" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(sample_id, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-498"><a href="#cb69-498" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(infection_status, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-499"><a href="#cb69-499" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(death_during_admission, <span class="at">new_role =</span> <span class="st">"outcome"</span>) <span class="sc">%&gt;%</span>  <span class="co"># id role tells the model not to actually incorporate it as a feature</span></span>
<span id="cb69-500"><a href="#cb69-500" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(condition, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> <span class="co"># specify target </span></span>
<span id="cb69-501"><a href="#cb69-501" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_dummy</span>(<span class="fu">all_nominal_predictors</span>()) <span class="sc">%&gt;%</span> <span class="co"># convert string to dummy variables</span></span>
<span id="cb69-502"><a href="#cb69-502" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_normalize</span>(<span class="fu">all_numeric_predictors</span>()) <span class="co"># normalize numeric features</span></span>
<span id="cb69-503"><a href="#cb69-503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-504"><a href="#cb69-504" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-505"><a href="#cb69-505" aria-hidden="true" tabindex="-1"></a>mortality_vs_survival_res <span class="ot">=</span> <span class="fu">our_model</span>(model_data,<span class="st">"death_during_admission"</span>,mortality_vs_survival_recipe)</span>
<span id="cb69-506"><a href="#cb69-506" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-507"><a href="#cb69-507" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-510"><a href="#cb69-510" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-511"><a href="#cb69-511" aria-hidden="true" tabindex="-1"></a><span class="fu">extract_roc</span>(mortality_vs_survival_res,<span class="st">"death_during_admission"</span>,<span class="st">".pred_FALSE"</span>)</span>
<span id="cb69-512"><a href="#cb69-512" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-513"><a href="#cb69-513" aria-hidden="true" tabindex="-1"></a>**Figure 12.** Here is a ROC curve of our model. It looks like it is reasonably </span>
<span id="cb69-514"><a href="#cb69-514" aria-hidden="true" tabindex="-1"></a>accurate, but note that there is a big class imbalance here. A simple model </span>
<span id="cb69-515"><a href="#cb69-515" aria-hidden="true" tabindex="-1"></a>where no patients die would get it right 76% of the time (57/75). </span>
<span id="cb69-516"><a href="#cb69-516" aria-hidden="true" tabindex="-1"></a>Our fancy model is only slightly slightly improved on this. </span>
<span id="cb69-517"><a href="#cb69-517" aria-hidden="true" tabindex="-1"></a>We are really hindered by our ability to only test a total of 3 true positives </span>
<span id="cb69-518"><a href="#cb69-518" aria-hidden="true" tabindex="-1"></a>(due to the small sample size).</span>
<span id="cb69-519"><a href="#cb69-519" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-522"><a href="#cb69-522" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-523"><a href="#cb69-523" aria-hidden="true" tabindex="-1"></a>mortality_vs_survival_res<span class="sc">$</span>confusion_matrix</span>
<span id="cb69-524"><a href="#cb69-524" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-525"><a href="#cb69-525" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-526"><a href="#cb69-526" aria-hidden="true" tabindex="-1"></a>**Table 1.** Here a confusion matrix highlights this class imbalance. </span>
<span id="cb69-527"><a href="#cb69-527" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-530"><a href="#cb69-530" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-531"><a href="#cb69-531" aria-hidden="true" tabindex="-1"></a>mortality_vs_survival_res<span class="sc">$</span>model_summary_plot</span>
<span id="cb69-532"><a href="#cb69-532" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-533"><a href="#cb69-533" aria-hidden="true" tabindex="-1"></a>**Figure 13.** Here we can see the features important for the model. Note the size of the intercept. </span>
<span id="cb69-534"><a href="#cb69-534" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-535"><a href="#cb69-535" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-536"><a href="#cb69-536" aria-hidden="true" tabindex="-1"></a><span class="fu">### Metadata alone model</span></span>
<span id="cb69-537"><a href="#cb69-537" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-538"><a href="#cb69-538" aria-hidden="true" tabindex="-1"></a>I am curious how well a model made with just the metadata would perform. </span>
<span id="cb69-539"><a href="#cb69-539" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-542"><a href="#cb69-542" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-543"><a href="#cb69-543" aria-hidden="true" tabindex="-1"></a>model_data <span class="ot">&lt;-</span> clinical_metadata <span class="sc">%&gt;%</span> </span>
<span id="cb69-544"><a href="#cb69-544" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(condition <span class="sc">!=</span> <span class="st">"healthy"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-545"><a href="#cb69-545" aria-hidden="true" tabindex="-1"></a>  <span class="co"># antimicrobial testing was not perfomred for this patient</span></span>
<span id="cb69-546"><a href="#cb69-546" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sample_id <span class="sc">!=</span> <span class="st">"S18"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-547"><a href="#cb69-547" aria-hidden="true" tabindex="-1"></a>  <span class="co">#getting rid of uninformative columns. Either very little variance, or too many levels to be useful </span></span>
<span id="cb69-548"><a href="#cb69-548" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>pathogen,<span class="sc">-</span>patient_id,<span class="sc">-</span>pathogen_source,<span class="sc">-</span>infection_status,</span>
<span id="cb69-549"><a href="#cb69-549" aria-hidden="true" tabindex="-1"></a>         <span class="sc">-</span>bacteremia_recurrence,<span class="sc">-</span>race,<span class="sc">-</span>death_at_one_year,</span>
<span id="cb69-550"><a href="#cb69-550" aria-hidden="true" tabindex="-1"></a>         <span class="sc">-</span>peripheral_vascular_disease,<span class="sc">-</span>dementia,<span class="sc">-</span>hemiplegia,</span>
<span id="cb69-551"><a href="#cb69-551" aria-hidden="true" tabindex="-1"></a>         <span class="sc">-</span>highly_active_antiretroviral_therapy,</span>
<span id="cb69-552"><a href="#cb69-552" aria-hidden="true" tabindex="-1"></a>         <span class="sc">-</span>source) <span class="sc">%&gt;%</span> </span>
<span id="cb69-553"><a href="#cb69-553" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">transplant_type =</span> <span class="fu">case_when</span>(transplant_type <span class="sc">==</span> <span class="st">"none"</span> <span class="sc">~</span> <span class="st">"none"</span>,</span>
<span id="cb69-554"><a href="#cb69-554" aria-hidden="true" tabindex="-1"></a>                                     <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"transplant"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb69-555"><a href="#cb69-555" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select_if</span>(<span class="sc">~</span> <span class="sc">!</span><span class="fu">any</span>(<span class="fu">is.na</span>(.))) <span class="sc">%&gt;%</span> </span>
<span id="cb69-556"><a href="#cb69-556" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>antibiotic_therapy_before_48_hrs,<span class="sc">-</span>antibiotic_therapy_after_48_hrs) <span class="sc">%&gt;%</span> </span>
<span id="cb69-557"><a href="#cb69-557" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(history_of_myocardial_infarction, </span>
<span id="cb69-558"><a href="#cb69-558" aria-hidden="true" tabindex="-1"></a>                   congestive_heart_failure, </span>
<span id="cb69-559"><a href="#cb69-559" aria-hidden="true" tabindex="-1"></a>                   history_of_stroke_or_transient_ischemic_attack, </span>
<span id="cb69-560"><a href="#cb69-560" aria-hidden="true" tabindex="-1"></a>                   smoking, copd, peptic_ulcer_disaese, </span>
<span id="cb69-561"><a href="#cb69-561" aria-hidden="true" tabindex="-1"></a>                   mod_to_severe_chromic_kidney_disease, </span>
<span id="cb69-562"><a href="#cb69-562" aria-hidden="true" tabindex="-1"></a>                   hodgkin_disease, leukemia_or_lymphoma, </span>
<span id="cb69-563"><a href="#cb69-563" aria-hidden="true" tabindex="-1"></a>                   icu_admission, mechanical_ventilation, </span>
<span id="cb69-564"><a href="#cb69-564" aria-hidden="true" tabindex="-1"></a>                    polymicrobial_bacteremia, </span>
<span id="cb69-565"><a href="#cb69-565" aria-hidden="true" tabindex="-1"></a>                   thrombocytopenia_plt_50, hypotension,sensitive_to_vancomycin,</span>
<span id="cb69-566"><a href="#cb69-566" aria-hidden="true" tabindex="-1"></a>                   death_during_admission), </span>
<span id="cb69-567"><a href="#cb69-567" aria-hidden="true" tabindex="-1"></a>                as.factor)) <span class="sc">%&gt;%</span> </span>
<span id="cb69-568"><a href="#cb69-568" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">wbc =</span> <span class="fu">as.numeric</span>(wbc))</span>
<span id="cb69-569"><a href="#cb69-569" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-570"><a href="#cb69-570" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-571"><a href="#cb69-571" aria-hidden="true" tabindex="-1"></a>mort_m_model_recipe <span class="ot">&lt;-</span> <span class="fu">recipe</span>(death_during_admission <span class="sc">~</span> ., <span class="at">data =</span> model_data) <span class="sc">%&gt;%</span></span>
<span id="cb69-572"><a href="#cb69-572" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(sample_id, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-573"><a href="#cb69-573" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(death_during_admission, <span class="at">new_role =</span> <span class="st">"outcome"</span>) <span class="sc">%&gt;%</span>  <span class="co"># id role tells the model not to actually incorporate it as a feature</span></span>
<span id="cb69-574"><a href="#cb69-574" aria-hidden="true" tabindex="-1"></a>  <span class="co">#update_role(condition, new_role = "outcome") %&gt;% # specify target </span></span>
<span id="cb69-575"><a href="#cb69-575" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_dummy</span>(<span class="fu">all_nominal_predictors</span>()) <span class="sc">%&gt;%</span> <span class="co"># convert string to dummy variables</span></span>
<span id="cb69-576"><a href="#cb69-576" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_impute_knn</span>(wbc) <span class="sc">%&gt;%</span> </span>
<span id="cb69-577"><a href="#cb69-577" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_normalize</span>(<span class="fu">all_numeric_predictors</span>()) <span class="co"># normalize numeric features</span></span>
<span id="cb69-578"><a href="#cb69-578" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-579"><a href="#cb69-579" aria-hidden="true" tabindex="-1"></a>mort_metadata_res <span class="ot">=</span> <span class="fu">our_model</span>(model_data,<span class="st">"death_during_admission"</span>,</span>
<span id="cb69-580"><a href="#cb69-580" aria-hidden="true" tabindex="-1"></a>                              mort_m_model_recipe)</span>
<span id="cb69-581"><a href="#cb69-581" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-582"><a href="#cb69-582" aria-hidden="true" tabindex="-1"></a><span class="fu">extract_roc</span>(mort_metadata_res,<span class="st">"death_during_admission"</span>,<span class="st">".pred_FALSE"</span>)</span>
<span id="cb69-583"><a href="#cb69-583" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-584"><a href="#cb69-584" aria-hidden="true" tabindex="-1"></a>**Figure 14.** Here we see that the clinical metadata does better job of predicting mortality than oru</span>
<span id="cb69-585"><a href="#cb69-585" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-586"><a href="#cb69-586" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-587"><a href="#cb69-587" aria-hidden="true" tabindex="-1"></a>What is going on here? </span>
<span id="cb69-588"><a href="#cb69-588" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-591"><a href="#cb69-591" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-592"><a href="#cb69-592" aria-hidden="true" tabindex="-1"></a>mort_metadata_res<span class="sc">$</span>model_summary_plot</span>
<span id="cb69-593"><a href="#cb69-593" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-594"><a href="#cb69-594" aria-hidden="true" tabindex="-1"></a>**Figure 15.* Turns out if you get admitted to ICU, that predicts mortality </span>
<span id="cb69-595"><a href="#cb69-595" aria-hidden="true" tabindex="-1"></a>pretty well (obviously). Once again, this is kind of cheating, ICU admission is </span>
<span id="cb69-596"><a href="#cb69-596" aria-hidden="true" tabindex="-1"></a>a feature that is only apparent after a patient is at the hospital for some time. </span>
<span id="cb69-597"><a href="#cb69-597" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-598"><a href="#cb69-598" aria-hidden="true" tabindex="-1"></a>What if we remove this variable from the model?</span>
<span id="cb69-599"><a href="#cb69-599" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-602"><a href="#cb69-602" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb69-603"><a href="#cb69-603" aria-hidden="true" tabindex="-1"></a>mort_m_model_recipe <span class="ot">&lt;-</span> <span class="fu">recipe</span>(death_during_admission <span class="sc">~</span> ., <span class="at">data =</span> model_data) <span class="sc">%&gt;%</span></span>
<span id="cb69-604"><a href="#cb69-604" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(sample_id, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb69-605"><a href="#cb69-605" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(death_during_admission, <span class="at">new_role =</span> <span class="st">"outcome"</span>) <span class="sc">%&gt;%</span>  <span class="co"># id role tells the model not to actually incorporate it as a feature</span></span>
<span id="cb69-606"><a href="#cb69-606" aria-hidden="true" tabindex="-1"></a>  <span class="fu">update_role</span>(icu_admission, <span class="at">new_role =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span> <span class="co"># specify target </span></span>
<span id="cb69-607"><a href="#cb69-607" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_dummy</span>(<span class="fu">all_nominal_predictors</span>()) <span class="sc">%&gt;%</span> <span class="co"># convert string to dummy variables</span></span>
<span id="cb69-608"><a href="#cb69-608" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_impute_knn</span>(wbc) <span class="sc">%&gt;%</span> </span>
<span id="cb69-609"><a href="#cb69-609" aria-hidden="true" tabindex="-1"></a>  <span class="fu">step_normalize</span>(<span class="fu">all_numeric_predictors</span>()) <span class="co"># normalize numeric features</span></span>
<span id="cb69-610"><a href="#cb69-610" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-611"><a href="#cb69-611" aria-hidden="true" tabindex="-1"></a>mort_metadata_res <span class="ot">=</span> <span class="fu">our_model</span>(model_data,<span class="st">"death_during_admission"</span>,</span>
<span id="cb69-612"><a href="#cb69-612" aria-hidden="true" tabindex="-1"></a>                              mort_m_model_recipe)</span>
<span id="cb69-613"><a href="#cb69-613" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-614"><a href="#cb69-614" aria-hidden="true" tabindex="-1"></a><span class="fu">extract_roc</span>(mort_metadata_res,<span class="st">"death_during_admission"</span>,<span class="st">".pred_FALSE"</span>)</span>
<span id="cb69-615"><a href="#cb69-615" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-616"><a href="#cb69-616" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb69-617"><a href="#cb69-617" aria-hidden="true" tabindex="-1"></a>**Figure 16.** If you take out the ICU admission, you lose all model performance.</span>
<span id="cb69-618"><a href="#cb69-618" aria-hidden="true" tabindex="-1"></a>You actually would be better off just predicting no one dies. </span>
<span id="cb69-619"><a href="#cb69-619" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-620"><a href="#cb69-620" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-621"><a href="#cb69-621" aria-hidden="true" tabindex="-1"></a><span class="fu"># Conclusion</span></span>
<span id="cb69-622"><a href="#cb69-622" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-623"><a href="#cb69-623" aria-hidden="true" tabindex="-1"></a><span class="ss">1. </span>The results of the lasso regression models show very similar results/ </span>
<span id="cb69-624"><a href="#cb69-624" aria-hidden="true" tabindex="-1"></a>interpretations to the other form of analysis that we conducted. </span>
<span id="cb69-625"><a href="#cb69-625" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-626"><a href="#cb69-626" aria-hidden="true" tabindex="-1"></a><span class="ss">2. </span>These analyses should adress Review 2 comments appropriately, </span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>